# MAG catalogue

```{r load_data_mag, message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

## Genome phylogeny

```{r genome_phylogeny, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate new species table
newspecies_table <- genome_metadata %>% 
  mutate(newspecies=ifelse(species=="s__","Y","N")) %>% 
  select(genome,newspecies) %>% 
  column_to_rownames(var = "genome")

# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.2)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.05, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.3,
                pwidth = 0.1,
                orientation="y",
              stat="identity")

# Add genome-size ring
circular_tree <- circular_tree + new_scale_fill()

circular_tree <- gheatmap(circular_tree, newspecies_table, offset=1.2, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#f4f4f4","#666666"))
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Add text
circular_tree <-  circular_tree +
        annotate('text', x=2.9, y=0, label='              Phylum', family='arial', size=3.5) +
        annotate('text', x=3.3, y=0, label='            Group', family='arial', size=3.5) +
        annotate('text', x=3.8, y=0, label='                         Genome quality', family='arial', size=3.5) +
        annotate('text', x=4.2, y=0, label='                     New species', family='arial', size=3.5)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

## Taxonomy overview

```{r genome_taxonomy_bacteria, message=FALSE, warning=FALSE,}
tax_mag <-genome_metadata %>% 
  group_by(phylum) %>%
  summarise(mag_n=n()) 

tax_mag %>%
  mutate(percetage_mag=round(mag_n*100/sum(mag_n), 2)) %>% 
  arrange(-percetage_mag) %>%
  tt()
```

## Mag size (MB)

```{r}
genome_metadata <-genome_metadata %>% 
  mutate(corrected_size=100*length/completeness)
```

Mags average size (MB)
```{r mag_size_all_mean, message=FALSE, warning=FALSE}
genome_metadata %>% 
  summarise(Average_corrected_size=mean(corrected_size)) %>% 
  tt()
```

Minimum Mags size (MB)

```{r mag_size_min,message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(corrected_size==min(corrected_size)) %>% 
  tt()
```

Maximum Mags size (MB)

```{r mag_size_max,message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(corrected_size==max(corrected_size)) %>% 
  tt()
```


Mags average size and completeness by phylum

```{r mag_size_phylum, message=FALSE, warning=FALSE}
genome_metadata %>% 
  group_by(phylum) %>%
  summarise(average_size=mean(corrected_size),sd_size=sd(corrected_size),average_comp=mean(completeness),sd_comp=sd(completeness)) %>%
  mutate(Size=str_c(round(average_size,2),"±",round(sd_size,2)),
         Completeness=str_c(round(average_comp,2),"±",round(sd_comp,2))) %>% 
  arrange(-average_size) %>% 
  select(phylum, Size, Completeness) %>% 
  tt()
```

Mags average size and completeness by phylum in wild

```{r mag_size_phylum_wild, message=FALSE, warning=FALSE}

wild_data <-sample_metadata %>% 
  filter(diet=="Wild")
  
  
wild_genome <- genome_counts %>% 
  select(genome, all_of(wild_data$sample)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) 
  

wild_genome_metadata <- genome_metadata %>% 
  filter(genome %in% wild_genome$genome) %>% 
  arrange(match(genome,wild_genome$genome))

wild_genome_metadata %>% 
  group_by(phylum) %>%
  summarise(average_size=mean(corrected_size),sd_size=sd(corrected_size),average_comp=mean(completeness),sd_comp=sd(completeness)) %>%
  mutate(Size=str_c(round(average_size,2),"±",round(sd_size,2)),
         Completeness=str_c(round(average_comp,2),"±",round(sd_comp,2))) %>% 
  arrange(-average_size) %>% 
  select(phylum, Size, Completeness) %>% 
  tt()

```

## Genome quality

```{r genome_quality, message=FALSE, warning=FALSE}
genome_metadata %>% 
    summarise(completeness_mean=mean(completeness) %>% round(2) %>% as.character(), 
              completeness_sd=sd(completeness) %>% round(2) %>% as.character(), 
              contamination_mean=mean(contamination) %>% round(2), 
              contamination_sd=sd(contamination) %>% round(2)) %>%
    unite("Completeness",completeness_mean, completeness_sd, sep = " ± ", remove = TRUE) %>%
    unite("Contamination",contamination_mean, contamination_sd, sep = " ± ", remove = TRUE) %>%
    tt()
```

```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_void() +
                    theme(legend.position = "none",
                        axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        axis.text.y=element_blank(),
                        axis.ticks.y=element_blank(),
                        axis.text.x=element_blank(),
                        axis.ticks.x=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(50,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_void() +
                theme(legend.position = "none",
                    axis.title.x = element_blank(),
                    axis.title.y = element_blank(),
                    axis.text.y=element_blank(),
                    axis.ticks.y=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))

```

## Functional overview

```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into elements
function_table <- genome_gifts %>%
    to.elements(., GIFT_db)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination

PCoA functional ordination with PCA loadings.

```{r function_ordination_pcoa, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
gift_pcoa <- genome_gifts %>%
    to.elements(., GIFT_db) %>%
    as.data.frame() %>%
    vegdist(method="euclidean") %>%
    pcoa()

gift_pcoa_rel_eigen <- gift_pcoa$values$Relative_eig[1:10]


# Get genome positions
gift_pcoa_vectors <- gift_pcoa$vectors %>% #extract vectors
  as.data.frame() %>% 
  select(Axis.1,Axis.2) # keep the first 2 axes

gift_pcoa_eigenvalues <- gift_pcoa$values$Eigenvalues[c(1,2)]

gift_pcoa_gifts <- cov(genome_gifts, scale(gift_pcoa_vectors)) %*% diag((gift_pcoa_eigenvalues/(nrow(genome_gifts)-1))^(-0.5)) %>%
  as.data.frame() %>% 
  rename(Axis.1=1,Axis.2=2) %>% 
  rownames_to_column(var="label") %>% 
  #get function summary vectors
  mutate(func=substr(label,1,3)) %>% 
  group_by(func) %>% 
  summarise(Axis.1=mean(Axis.1),
            Axis.2=mean(Axis.2)) %>% 
  rename(label=func) %>% 
  filter(!label %in% c("S01","S02","S03"))

```


```{r function_ordination_pcoa_plot, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

scale <- 15 # scale for vector loadings
gift_pcoa_vectors %>% 
  rownames_to_column(var="genome") %>% 
  left_join(genome_metadata, by="genome") %>%
  ggplot() +
      #genome positions
      scale_color_manual(values=phylum_colors)+
      geom_point(aes(x=Axis.1,y=Axis.2, color=phylum, size=length), 
                 alpha=0.9, shape=16) +
      #scale_color_manual(values=phylum_colors) +
      scale_size_continuous(range = c(0.1,5)) +
      #loading positions
      geom_segment(data=gift_pcoa_gifts, 
                   aes(x=0, y=0, xend=Axis.1 * scale, yend=Axis.2 * scale),
                    arrow = arrow(length = unit(0.3, "cm"), 
                    type = "open", 
                    angle = 25),
                    linewidth = 0.5, 
                    color = "black") +
     #Primary and secondary scale adjustments
     scale_x_continuous(name = paste0("PCoA1 (",round(gift_pcoa_rel_eigen[1]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA1")
            ) +
     scale_y_continuous(name = paste0("PCoA2 (",round(gift_pcoa_rel_eigen[2]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA2")
            ) +
    geom_label_repel(data = gift_pcoa_gifts,
                     aes(label = label, x = Axis.1 * scale, y = Axis.2 * scale),
                     segment.color = 'transparent') +
    theme_minimal() + 
    theme(legend.position = "none")

```

```{r function_ordination_pcoa_plot_treatment_time_mci, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>% 
  filter(!is.na(day)) %>% 
  group_by(genome,treatment,day) %>% 
  summarise(count=mean(count)) %>% 
  left_join(gift_pcoa_vectors %>% rownames_to_column(var="genome"), by="genome") %>%
  left_join(genome_metadata, by="genome") %>%
  ggplot(aes(x=Axis.1, y=count, group=treatment, color=treatment)) +
        scale_color_manual(values=treatment_colors)+
        geom_smooth(method = lm, formula = y ~ splines::bs(x, 3), se = FALSE) +
        facet_grid(. ~ day) +
        theme_minimal() + 
        theme(legend.position = "none")
```

```{r function_ordination_pcoa_plot_treatment_time_functions, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genome_gifts %>%
    to.elements(., GIFT_db) %>% 
    to.functions(., GIFT_db) %>% 
    to.community(., genome_counts_filt %>%
                    mutate_at(vars(-genome),~./sum(.)) %>%
                   column_to_rownames(var="genome"), GIFT_db) %>% 
    as.data.frame() %>% 
    rownames_to_column(var="sample") %>% 
    pivot_longer(-sample, names_to = "trait", values_to = "value") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>% 
    group_by(trait,treatment,day) %>% 
    summarise(value=mean(value)) %>% 
    mutate(trait=factor(trait)) %>% 
    filter(treatment != "TG0") %>% 
    ggplot(aes(x=fct_reorder(trait, desc(value)), y=value, fill=trait)) +
        geom_bar(stat="identity", width=1) +
    facet_grid(treatment ~ day) +
    theme_minimal() + 
    theme(legend.position = "none")
```

```{r function_ordination_pcoa_plot_treatment_time_mci2, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genome_gifts %>%
    to.elements(., GIFT_db) %>% 
    to.functions(., GIFT_db) %>% 
    to.community(., genome_counts_filt %>%
                    mutate_at(vars(-genome),~./sum(.)) %>%
                   column_to_rownames(var="genome"), GIFT_db) %>% 
    as.data.frame() %>% 
    rownames_to_column(var="sample") %>% 
    pivot_longer(-sample, names_to = "trait", values_to = "value") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>% 
    group_by(sample,treatment,day) %>% 
    summarise(value=mean(value)) %>% 
    filter(treatment != "TG0") %>% 
    ggplot(aes(x=day, y=value, group=day, fill=treatment, color=treatment)) +
        geom_boxplot() +
        geom_jitter() +
        scale_color_manual(name="Treatment",
          breaks=c("TG1","TG2","TG3","TG4","TG5"),
          values=c("#4059AE", "#6A9AC3","#97D8C4","#AFD699","#F3B942")) +
      scale_fill_manual(name="Treatment",
          breaks=c("TG1","TG2","TG3","TG4","TG5"),
          values=c("#4059AE50", "#6A9AC350","#97D8C450","#AFD69950","#F3B94250")) +
    facet_grid(treatment ~ .) +
    theme_linedraw() + 
    theme(legend.position = "none")
```