---
title: "AlberdiLab | Brenner et al. 2025"
subtitle: "Sled dogs extended"
author:
  - Elsa Brenner, Antton Alberdi, Ostaizka Aizpurua^[University of Copenhagen, ostaizka.aizpurua@sund.ku.dk]
date: "Last update: `r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
url: https://alberdilab.github.io/sled_dogs_extended
description: |
  Data analysis code for the study on the study of gut microbiomes of sled dogs and beyond.
link-citations: yes
github-repo: alberdilab/sled_dogs_extended
---

```{r knitr_opts, echo=FALSE}
knitr::opts_chunk$set(
    class.source = "script-source",
    class.output = "script-output",
    comment = NA)
```

# Introduction

This webbook contains all the code used for data analysis in study of gut microbiomes of newts across ponds included in a restoration plan.

## Prepare the R environment

### Environment

To reproduce all the analyses locally, clone this repository in your computer using:

```
RStudio > New Project > Version Control > Git
```

And indicating the following git repository:

> https://github.com/alberdilab/sled_dogs_extended.git

Once the R project has been created, follow the instructions and code chunks shown in this webbook.

### Libraries

The following R packages are required for the data analysis.

```{r load_libraries, warning=FALSE, comments="", message=FALSE}
# Base
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(rairtable)
library(janitor)
library(broom)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)
library(UpSetR)
library(viridis)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(ANCOMBC)
library(lme4)
library(nlme)
library(pairwiseAdonis)
library(emmeans)
library(pheatmap)
library(rstatix)
```

<!--chapter:end:index.Rmd-->

# Prepare data

Load the original data files outputted by the bioinformatic pipeline.

### Reads
```{r load_reads, warning=FALSE, comments="", message=FALSE, eval=FALSE}
read_counts <- read_tsv("https://sid.erda.dk/share_redirect/AvhSMDyzyU/counts.tsv") %>%
    rename(genome=1) %>% 
    rename_with(~ str_remove(.x, "_EKDL.*")) %>% 
    rename_with(~ str_remove(.x, "_M$"))
```

### Sample metadata
```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_metadata <- read_csv("data/sample_metadata.csv")
```


### Filter read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE, eval=FALSE}
read_counts <- read_counts %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
    select(one_of(c("genome",sample_metadata$sample)))
```

### Genome taxonomy

```{r load_genome_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_taxonomy <- read_tsv("https://sid.erda.dk/share_redirect/AvhSMDyzyU/genome_taxonomy.tsv") %>% 
  rename(genome = user_genome) %>% 
  separate(
    classification,
    into  = c("domain", "phylum", "class", "order", "family", "genus","species"),
    sep   = ";",          # split on the semicolons
    fill  = "right",      # if a rank is missing, keep NA on the right
    remove = TRUE
  ) %>%
  mutate(across(domain:species, ~ str_remove(., "^[a-z]__"))) %>% 
  select(genome,domain,phylum,class,order,family,genus,species)

genome_features <- read_csv("https://sid.erda.dk/share_redirect/AvhSMDyzyU/genomeInfo.csv") %>% 
  mutate(genome = str_remove(genome, "\\.fa$"))

genome_metadata <- inner_join(genome_taxonomy,genome_features, by="genome")
```

### Genome base hits

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_coverage <- read_tsv("https://sid.erda.dk/share_redirect/AvhSMDyzyU/bases.tsv") %>%
    rename(genome=1) %>% 
    rename_with(~ str_remove(.x, "_EKDL.*")) %>% 
    rename_with(~ str_remove(.x, "_M$"))
```


### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_tree <- read_tree("https://sid.erda.dk/share_redirect/AvhSMDyzyU/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips
```

### Genome annotations

```{r load_genome_annotation, warning=FALSE, comments="", message=FALSE, eval=FALSE}
download.file("https://sid.erda.dk/share_redirect/AvhSMDyzyU/gene_annotations.tsv.xz", destfile = "data/gene_annotations.tsv.xz", mode = "wb")
genome_annotations <- read_tsv("data/gene_annotations.tsv.xz") %>% 
  mutate(genome = str_extract(gene, "^[^^]+")) %>% 
  mutate(ec = str_c("[EC:",ec,"]"))
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE, eval=FALSE}
min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]]))
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE, eval=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE, eval=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs

```{r distill_annotations, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_gifts <- distill(genome_annotations,GIFT_db,genomecol=14,annotcol=c(5,6), verbosity=F)
genome_gifts <- genome_gifts[, !grepl("^S", colnames(genome_gifts))]
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE, eval=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    mutate(phylum = str_remove(phylum, "^p__")) %>% 
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>%
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)
```


## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(sample_metadata,
     genome_metadata,
     read_counts,
     genome_counts,
     genome_counts_filt,
     genome_tree,
     genome_gifts,
     phylum_colors,
     file = "data/data.Rdata")
```

<!--chapter:end:01_prepare_data.Rmd-->

# Data statistics

```{r load_data_stats}
load("data/data.Rdata")
```

## filter samples with high host data
```{r load_data_host_filt_host, comment="", message=FALSE, warning=FALSE, eval=FALSE}
sample_metadata <- sample_metadata%>%
  filter(!sample %in% c("EHI02721", "EHI02712", "EHI02700", "EHI02720", "EHI02749", "EHI02719", "EHI02729", "EHI02715", "EHI02722"))

read_counts <- read_counts %>% 
    select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., read_counts, by = "genome") %>% 
  arrange(match(genome,read_counts$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips

#load("data/genome_gifts.Rdata")
```

## Sequencing reads statistics

```{r reads_stats}
sample_metadata %>% 
    summarise(Total=sum(reads_post_fastp * 150 / 1000000000) %>% round(2), 
              mean=mean(reads_post_fastp * 150 / 1000000000) %>% round(2),
              sd=sd(reads_post_fastp * 150 / 1000000000) %>% round(2)) %>%
    unite("Average",mean, sd, sep = " ± ", remove = TRUE) %>%
    tt()
```

## DNA fractions
```{r dna_fractions_stats}
sequence_fractions <- read_counts %>%
  pivot_longer(-genome, names_to = "sample", values_to = "value") %>%
  group_by(sample) %>%
  summarise(mags = sum(value)) %>%
	left_join(sample_metadata, by = join_by(sample == sample)) %>%
	select(sample,mags,metagenomic_bases,host_bases,bases_lost_fastp_percent) %>%
	mutate(mags_bases = mags*146) %>%
	mutate(lowqual_bases = ((metagenomic_bases+host_bases)/(1-bases_lost_fastp_percent))-(metagenomic_bases+host_bases)) %>%
	mutate(unmapped_bases = metagenomic_bases - mags_bases) %>%
	mutate(unmapped_bases = ifelse(unmapped_bases < 0, 0, unmapped_bases)) %>%
	select(sample, lowqual_bases, host_bases, unmapped_bases, mags_bases)

sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  tt()

sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  summarise(across(where(is.numeric), mean, na.rm = TRUE)) %>%
  tt()

```


```{r dna_fractions_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
sequence_fractions %>%
	pivot_longer(!sample, names_to = "fraction", values_to = "value") %>%
	mutate(value = value / 1000000000) %>%
	mutate(fraction = factor(fraction, levels = c("lowqual_bases","host_bases","unmapped_bases","mags_bases"))) %>%
  
	ggplot(., aes(x = sample, y = value, fill=fraction)) +
	    geom_bar(position="stack", stat = "identity") +
      scale_fill_manual(name="Sequence type",
                    breaks=c("lowqual_bases","host_bases","unmapped_bases","mags_bases"),
                    labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                    values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
	    labs(x = "Samples", y = "Amount of data (GB)") +
	    theme_classic() +
	    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size=6),legend.position = "bottom")
```

## Recovered microbial fraction

```{r data_estimations_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
singlem_table <- sequence_fractions %>%
	mutate(mags_proportion = round((mags_bases / (mags_bases + unmapped_bases))*100,2)) %>%
  left_join(sample_metadata, by = join_by(sample == sample))  %>%
	mutate(singlem_proportion = round(singlem_fraction,2)) %>%
	select(sample,mags_proportion,singlem_proportion) %>%
	mutate(mags_proportion = ifelse(singlem_proportion == 0, 0, mags_proportion)) %>% #convert zeros to NA
	mutate(singlem_proportion = ifelse(singlem_proportion == 0, NA, singlem_proportion)) %>% #convert zeros to NA
	mutate(singlem_proportion = ifelse(singlem_proportion < mags_proportion, NA, singlem_proportion)) %>% #if singlem is smaller, then NA, to simplify plot
	mutate(singlem_proportion = ifelse(singlem_proportion > 100, 100, singlem_proportion)) #simplify

singlem_table %>%
	pivot_longer(!sample, names_to = "proportion", values_to = "value") %>%
	mutate(proportion = factor(proportion, levels = c("mags_proportion","singlem_proportion"))) %>%
	ggplot(., aes(x = value, y = sample, color=proportion)) +
			geom_line(aes(group = sample), color = "#f8a538") +
			geom_point() +
      scale_color_manual(name="Proportion",
                    breaks=c("mags_proportion","singlem_proportion"),
                    labels=c("Recovered","Estimated"),
                    values=c("#52e1e8", "#876b53"))+
			theme_classic() +
			labs(y = "Samples", x = "Prokaryotic fraction (%)") +
	    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size=6),legend.position = "right")

```

```{r damr, message=FALSE, warning=FALSE}
damr <- singlem_table %>%
  mutate(damr=ifelse(is.na(singlem_proportion),100,mags_proportion/singlem_proportion*100)) %>%
  select(sample,damr)

damr %>% 
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(diet=="Wild")%>% 
  summarise(mean=mean(damr),sd=sd(damr)) %>% 
  tt()

damr %>% 
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(region=="Nafarroa")%>% 
  summarise(mean=mean(damr),sd=sd(damr)) %>% 
  tt()

damr %>% 
  tt()

damr %>% 
  filter(damr>95) %>%
  nrow()

damr %>% 
  summarise(mean=mean(damr),sd=sd(damr)) %>% 
  tt()
```


<!--chapter:end:02_data_statistics.Rmd-->

# MAG catalogue

```{r load_data_mag, message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

## Genome phylogeny

```{r genome_phylogeny, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate new species table
newspecies_table <- genome_metadata %>% 
  mutate(newspecies=ifelse(species=="","Y","N")) %>% 
  select(genome,newspecies) %>% 
  column_to_rownames(var = "genome")

# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.2)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.05, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.15,
                pwidth = 0.1,
                orientation="y",
              stat="identity")

# Add genome-size ring
circular_tree <- circular_tree + new_scale_fill()

circular_tree <- gheatmap(circular_tree, newspecies_table, offset=0.5, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#f4f4f4","#666666"))
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Add text
circular_tree <-  circular_tree +
        annotate('text', x=2.2, y=0, label='              Phylum', family='arial', size=3.5) +
        annotate('text', x=2.4, y=0, label='                         Genome quality', family='arial', size=3.5) +
        annotate('text', x=2.6, y=0, label='                     New species', family='arial', size=3.5)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

## Taxonomy overview

```{r genome_taxonomy_bacteria, message=FALSE, warning=FALSE,}
tax_mag <-genome_metadata %>% 
  group_by(phylum) %>%
  summarise(mag_n=n()) 

tax_mag %>%
  mutate(percetage_mag=round(mag_n*100/sum(mag_n), 2)) %>% 
  arrange(-percetage_mag) %>%
  tt()
```

## Mag size (MB)

```{r}
genome_metadata <-genome_metadata %>% 
  mutate(corrected_size=100*length/completeness)
```

Mags average size (MB)
```{r mag_size_all_mean, message=FALSE, warning=FALSE}
genome_metadata %>% 
  summarise(Average_corrected_size=mean(corrected_size)) %>% 
  tt()
```

Minimum Mags size (MB)

```{r mag_size_min,message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(corrected_size==min(corrected_size)) %>% 
  tt()
```

Maximum Mags size (MB)

```{r mag_size_max,message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(corrected_size==max(corrected_size)) %>% 
  tt()
```


Mags average size and completeness by phylum

```{r mag_size_phylum, message=FALSE, warning=FALSE}
genome_metadata %>% 
  group_by(phylum) %>%
  summarise(average_size=mean(corrected_size),sd_size=sd(corrected_size),average_comp=mean(completeness),sd_comp=sd(completeness)) %>%
  mutate(Size=str_c(round(average_size,2),"±",round(sd_size,2)),
         Completeness=str_c(round(average_comp,2),"±",round(sd_comp,2))) %>% 
  arrange(-average_size) %>% 
  select(phylum, Size, Completeness) %>% 
  tt()
```

Mags average size and completeness by phylum in wild

```{r mag_size_phylum_wild, message=FALSE, warning=FALSE}

wild_data <-sample_metadata %>% 
  filter(diet=="Wild")
  
  
wild_genome <- genome_counts %>% 
  select(genome, all_of(wild_data$sample)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) 
  

wild_genome_metadata <- genome_metadata %>% 
  filter(genome %in% wild_genome$genome) %>% 
  arrange(match(genome,wild_genome$genome))

wild_genome_metadata %>% 
  group_by(phylum) %>%
  summarise(average_size=mean(corrected_size),sd_size=sd(corrected_size),average_comp=mean(completeness),sd_comp=sd(completeness)) %>%
  mutate(Size=str_c(round(average_size,2),"±",round(sd_size,2)),
         Completeness=str_c(round(average_comp,2),"±",round(sd_comp,2))) %>% 
  arrange(-average_size) %>% 
  select(phylum, Size, Completeness) %>% 
  tt()

```

## Genome quality

```{r genome_quality, message=FALSE, warning=FALSE}
genome_metadata %>% 
    summarise(completeness_mean=mean(completeness) %>% round(2) %>% as.character(), 
              completeness_sd=sd(completeness) %>% round(2) %>% as.character(), 
              contamination_mean=mean(contamination) %>% round(2), 
              contamination_sd=sd(contamination) %>% round(2)) %>%
    unite("Completeness",completeness_mean, completeness_sd, sep = " ± ", remove = TRUE) %>%
    unite("Contamination",contamination_mean, contamination_sd, sep = " ± ", remove = TRUE) %>%
    tt()
```

```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_void() +
                    theme(legend.position = "none",
                        axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        axis.text.y=element_blank(),
                        axis.ticks.y=element_blank(),
                        axis.text.x=element_blank(),
                        axis.ticks.x=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(50,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_void() +
                theme(legend.position = "none",
                    axis.title.x = element_blank(),
                    axis.title.y = element_blank(),
                    axis.text.y=element_blank(),
                    axis.ticks.y=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))

```

## Functional overview

```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into elements
function_table <- genome_gifts %>%
    to.elements(., GIFT_db)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination

PCoA functional ordination with PCA loadings.

```{r function_ordination_pcoa, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
gift_pcoa <- genome_gifts %>%
    to.elements(., GIFT_db) %>%
    as.data.frame() %>%
    vegdist(method="euclidean") %>%
    pcoa()

gift_pcoa_rel_eigen <- gift_pcoa$values$Relative_eig[1:10]


# Get genome positions
gift_pcoa_vectors <- gift_pcoa$vectors %>% #extract vectors
  as.data.frame() %>% 
  select(Axis.1,Axis.2) # keep the first 2 axes

gift_pcoa_eigenvalues <- gift_pcoa$values$Eigenvalues[c(1,2)]

gift_pcoa_gifts <- cov(genome_gifts, scale(gift_pcoa_vectors)) %*% diag((gift_pcoa_eigenvalues/(nrow(genome_gifts)-1))^(-0.5)) %>%
  as.data.frame() %>% 
  rename(Axis.1=1,Axis.2=2) %>% 
  rownames_to_column(var="label") %>% 
  #get function summary vectors
  mutate(func=substr(label,1,3)) %>% 
  group_by(func) %>% 
  summarise(Axis.1=mean(Axis.1),
            Axis.2=mean(Axis.2)) %>% 
  rename(label=func) %>% 
  filter(!label %in% c("S01","S02","S03"))

```


```{r function_ordination_pcoa_plot, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

scale <- 15 # scale for vector loadings
gift_pcoa_vectors %>% 
  rownames_to_column(var="genome") %>% 
  left_join(genome_metadata, by="genome") %>%
  ggplot() +
      #genome positions
      scale_color_manual(values=phylum_colors)+
      geom_point(aes(x=Axis.1,y=Axis.2, color=phylum, size=length), 
                 alpha=0.9, shape=16) +
      #scale_color_manual(values=phylum_colors) +
      scale_size_continuous(range = c(0.1,5)) +
      #loading positions
      geom_segment(data=gift_pcoa_gifts, 
                   aes(x=0, y=0, xend=Axis.1 * scale, yend=Axis.2 * scale),
                    arrow = arrow(length = unit(0.3, "cm"), 
                    type = "open", 
                    angle = 25),
                    linewidth = 0.5, 
                    color = "black") +
     #Primary and secondary scale adjustments
     scale_x_continuous(name = paste0("PCoA1 (",round(gift_pcoa_rel_eigen[1]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA1")
            ) +
     scale_y_continuous(name = paste0("PCoA2 (",round(gift_pcoa_rel_eigen[2]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA2")
            ) +
    geom_label_repel(data = gift_pcoa_gifts,
                     aes(label = label, x = Axis.1 * scale, y = Axis.2 * scale),
                     segment.color = 'transparent') +
    theme_minimal() + 
    theme(legend.position = "none")

```

```{r function_ordination_pcoa_plot_treatment_time_mci, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>% 
  filter(!is.na(day)) %>% 
  group_by(genome,treatment,day) %>% 
  summarise(count=mean(count)) %>% 
  left_join(gift_pcoa_vectors %>% rownames_to_column(var="genome"), by="genome") %>%
  left_join(genome_metadata, by="genome") %>%
  ggplot(aes(x=Axis.1, y=count, group=treatment, color=treatment)) +
        scale_color_manual(values=treatment_colors)+
        geom_smooth(method = lm, formula = y ~ splines::bs(x, 3), se = FALSE) +
        facet_grid(. ~ day) +
        theme_minimal() + 
        theme(legend.position = "none")
```

```{r function_ordination_pcoa_plot_treatment_time_functions, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genome_gifts %>%
    to.elements(., GIFT_db) %>% 
    to.functions(., GIFT_db) %>% 
    to.community(., genome_counts_filt %>%
                    mutate_at(vars(-genome),~./sum(.)) %>%
                   column_to_rownames(var="genome"), GIFT_db) %>% 
    as.data.frame() %>% 
    rownames_to_column(var="sample") %>% 
    pivot_longer(-sample, names_to = "trait", values_to = "value") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>% 
    group_by(trait,treatment,day) %>% 
    summarise(value=mean(value)) %>% 
    mutate(trait=factor(trait)) %>% 
    filter(treatment != "TG0") %>% 
    ggplot(aes(x=fct_reorder(trait, desc(value)), y=value, fill=trait)) +
        geom_bar(stat="identity", width=1) +
    facet_grid(treatment ~ day) +
    theme_minimal() + 
    theme(legend.position = "none")
```

```{r function_ordination_pcoa_plot_treatment_time_mci2, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genome_gifts %>%
    to.elements(., GIFT_db) %>% 
    to.functions(., GIFT_db) %>% 
    to.community(., genome_counts_filt %>%
                    mutate_at(vars(-genome),~./sum(.)) %>%
                   column_to_rownames(var="genome"), GIFT_db) %>% 
    as.data.frame() %>% 
    rownames_to_column(var="sample") %>% 
    pivot_longer(-sample, names_to = "trait", values_to = "value") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>% 
    group_by(sample,treatment,day) %>% 
    summarise(value=mean(value)) %>% 
    filter(treatment != "TG0") %>% 
    ggplot(aes(x=day, y=value, group=day, fill=treatment, color=treatment)) +
        geom_boxplot() +
        geom_jitter() +
        scale_color_manual(name="Treatment",
          breaks=c("TG1","TG2","TG3","TG4","TG5"),
          values=c("#4059AE", "#6A9AC3","#97D8C4","#AFD699","#F3B942")) +
      scale_fill_manual(name="Treatment",
          breaks=c("TG1","TG2","TG3","TG4","TG5"),
          values=c("#4059AE50", "#6A9AC350","#97D8C450","#AFD69950","#F3B94250")) +
    facet_grid(treatment ~ .) +
    theme_linedraw() + 
    theme(legend.position = "none")
```

<!--chapter:end:03_mag_catalogue.Rmd-->

# Community composition

## Filter data

```{r load_data_community, comment="", message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")

sample_metadata <- read_csv("data/sample_metadata.csv")
 
sample_metadata$individual <- sub("[0-9]{2}$", "", sample_metadata$animal)

sample_metadata_illu <- read_csv("data/metadata_illu.csv")
sample_metadata <- sample_metadata %>% 
  left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) 

load("data/beta.Rdata")
#samples to remove
samples_to_remove <- sample_metadata %>% 
  filter(individual == "GL_02_" | Samples %in% c("Blank", "Saliva") | group2 %in% c("Unknown") | season=="winter" ) %>% #!day %in% c("day_1", "Unknown")
  select(sample) %>% 
  pull()

sample_metadata <- sample_metadata %>% 
  filter(!sample %in% samples_to_remove)
  #filter(individual != "GL_02_" | !Samples %in% c("Blank", "Saliva") | group2 != "Unknown" | season!="winter") 

genome_counts_filt <- genome_counts_filt %>% 
  select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome)
```

## Taxonomy overview 

### Stacked barplot

```{r taxonomy_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
  facet_grid(~ country, scale="free", space = "free")+
 #   facet_nested(. ~ region+diet,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          strip.background = element_rect(fill = "white"),
    strip.text = element_text(size = 12, lineheight = 0.6),
    strip.placement = "outside",
    axis.text.x = element_blank(), axis.ticks.x = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

**Number of bacteria phyla**

```{r phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**Bacteria phyla in wild individuals**

```{r phyla_nat, comment="", echo=FALSE, message=FALSE, warning=FALSE}

wild_samples <- sample_metadata %>% 
  filter(region=="Gipuzkoa") %>% 
  dplyr::select(sample) %>% 
  pull()

wild_genomes <- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(wild_samples)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% wild_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```

**Bacteria phyla captive animals**

```{r phyla_cap, comment="", echo=FALSE, message=FALSE, warning=FALSE}

captive_samples <- sample_metadata %>% 
  filter(region=="Nafarroa") %>% 
  dplyr::select(sample) %>% 
  pull()

captive_genomes <- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(captive_samples)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% captive_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```

**Bacteria phyla before grass is included in the diet**

```{r phyla_pre, comment="", echo=FALSE, message=FALSE, warning=FALSE}

captive_pre_samples <- sample_metadata %>% 
  filter(diet=="Sled_dog") %>% 
  dplyr::select(sample) %>% 
  pull()

captive_pre_genomes <- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(captive_pre_samples)) %>%
  as.data.frame() %>% 
  mutate(row_sum = rowSums(.)) %>%
  filter(row_sum != 0) %>%
  select(-row_sum)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% captive_pre_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```

**Bacteria phyla after grass is included in the diet**

```{r phyla_post, comment="", echo=FALSE, message=FALSE, warning=FALSE}
captive_post_samples <- sample_metadata %>% 
  filter(diet=="Post_grass") %>% 
  dplyr::select(sample) %>% 
  pull()

captive_post_genomes <- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(captive_post_samples)) %>%
  as.data.frame() %>% 
  mutate(row_sum = rowSums(.)) %>%
  filter(row_sum != 0) %>%
  select(-row_sum)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% captive_post_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```

**Number of Archaea phyla**

```{r arch, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**Archaea phyla in wild individuals**

```{r arch_nat, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(genome %in% wild_genomes) %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**Archaea phyla before grass is included in the diet**

```{r arch_pre, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(genome %in% captive_pre_genomes) %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull()
```

**Archaea phyla after grass is included in the diet**

```{r arch_post, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(genome %in% captive_post_genomes) %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull()
```

### Genus and species annotation

**Number of MAGs without species-level annotation**
```{r nonspe, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>%
  filter(species == "s__") %>%
  summarize(Mag_nospecies = n())%>%
  select(Mag_nospecies) %>% 
  pull()

```
```{r nonspe_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
total_mag_phylum <- genome_metadata %>%
  group_by(phylum) %>%
  summarize(count_total = n())
genome_metadata %>%
  filter(species == "s__") %>%
  group_by(phylum) %>%
  summarize(count_nospecies = n()) %>% 
  left_join(total_mag_phylum, by = join_by(phylum == phylum)) %>% 
  mutate(percentage=100*count_nospecies/count_total) %>% 
  tt()

```

**Percentage of MAGs without species-level annotation**
```{r sp_percet, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nmags <- nrow(genome_counts)
nonspecies <- genome_metadata %>%
    filter(species == "s__") %>%
    nrow()
perct <- nonspecies*100/nmags
perct
```

**Number of MAGs without genera-level annotation**
```{r nongenera, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nongenera <- genome_metadata %>%
    filter(genus == "g__") %>%
    nrow()
cat(nongenera)
```


### Phylum relative abundances

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum,group2) %>%
  summarise(relabun=sum(count))
```

#### Origin: Dogs vs Sled dogs vs Wolves
```{r taxonomy_phylum_summary_origin, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              Dog_mean=mean(relabun[group2=="Dog"]*100, na.rm=T),
              Dog_sd=sd(relabun[group2=="Dog"]*100, na.rm=T),
              Sled_dog_mean=mean(relabun[group2=="Sled_dog"]*100, na.rm=T),
              Sled_dog_sd=sd(relabun[group2=="Sled_dog"]*100, na.rm=T),
              Wolf_mean=mean(relabun[group2=="Wolf"]*100, na.rm=T),
              Wolf_sd=sd(relabun[group2=="Wolf"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Dog=str_c(round(Dog_mean,3),"±",round(Dog_sd,3)),
           Sled_dog=str_c(round(Sled_dog_mean,3),"±",round(Sled_dog_sd,3)),
           Wolf=str_c(round(Wolf_mean,3),"±",round(Wolf_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total,Dog,Sled_dog,Wolf)
```

## Taxonomy boxplot

### Family

```{r taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,family, group2) %>%
  summarise(relabun=sum(count))
family_summary$group2 <- factor(family_summary$group2, levels=c("Sled_dog", "Dog", "Wolf"))
```

#### Origin: Dog vs Captive
```{r taxonomy_family_summary_origin, warning=FALSE, comments="", message=FALSE}
family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              Dog_mean=mean(relabun[group2=="Dog"]*100, na.rm=T),
              Dog_sd=sd(relabun[group2=="Dog"]*100, na.rm=T),
              Sled_dog_mean=mean(relabun[group2=="Sled_dog"]*100, na.rm=T),
              Sled_dog_sd=sd(relabun[group2=="Sled_dog"]*100, na.rm=T),
              Wolf_mean=mean(relabun[group2=="Wolf"]*100, na.rm=T),
              Wolf_sd=sd(relabun[group2=="Wolf"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Dog=str_c(round(Dog_mean,3),"±",round(Dog_sd,3)),
           Sled_dog=str_c(round(Sled_dog_mean,3),"±",round(Sled_dog_sd,3)),
           Wolf=str_c(round(Wolf_mean,3),"±",round(Wolf_sd,3))) %>% 
    arrange(-Sled_dog_mean) %>% 
    dplyr::select(family,total,Dog,Sled_dog,Wolf)
```

### Genus

```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,phylum,genus, group2) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__") %>%
  mutate(genus= sub("^g__", "", genus))
genus_summary$group2 <- factor(genus_summary$group2, levels=c("Sled_dog", "Wolf", "Dog"))
```
```{r}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% 
  filter(genome=="ERR1915648_bin_35072")%>% #append genome metadata
  group_by(group2) %>%
  summarise(relabun=sum(count))
```
```{r}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% 
  filter(genus=="Cetobacterium_A")%>% #append genome metadata
  group_by(genome, group2) %>%
  summarise(relabun=sum(count), sd=sd(count))
```

```{r taxonomy_genera_summary_origroup2, warning=FALSE, comments="", message=FALSE}
genus_summary %>%
    group_by(genus) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              Dog_mean=mean(relabun[group2=="Dog"]*100, na.rm=T),
              Dog_sd=sd(relabun[group2=="Dog"]*100, na.rm=T),
              Sled_dog_mean=mean(relabun[group2=="Sled_dog"]*100, na.rm=T),
              Sled_dog_sd=sd(relabun[group2=="Sled_dog"]*100, na.rm=T),
              Wolf_mean=mean(relabun[group2=="Wolf"]*100, na.rm=T),
              Wolf_sd=sd(relabun[group2=="Wolf"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Dog=str_c(round(Dog_mean,3),"±",round(Dog_sd,3)),
           Sled_dog=str_c(round(Sled_dog_mean,3),"±",round(Sled_dog_sd,3)),
           Wolf=str_c(round(Wolf_mean,3),"±",round(Wolf_sd,3))) %>% 
    arrange(genus) %>% 
    dplyr::select(genus,total,Dog,Sled_dog,Wolf)
```

```{r taxonomy_jitterplot_genus, fig.height=14, fig.width=10, fig.fullwidth=TRUE}
genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary_sort <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=T),sd=sd(relabun, na.rm=T)) %>%
    arrange(-mean) 

genus_summary %>%
  mutate(genus=factor(genus, levels=rev(genus_summary_sort %>% pull(genus)))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
  scale_color_manual(values=phylum_colors) +
  geom_jitter(alpha=0.5) + 
  facet_grid(.~group2)+
  theme_minimal() + 
  theme(axis.text.y = element_text(size=6))+
  labs(y="Family", x="Relative abundance", color="Phylum")

```


<!--chapter:end:04_community_composition.Rmd-->

# Alpha diversity

```{r load_data_alpha, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

## Summary table

```{r alpha_div, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))
```

### Plots

```{r alpha_div_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = country, group=country, color=country, fill=country)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = country, group=country, color=country, fill=country)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = country, group=country, color=country, fill=country)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Phylogenetic")
```

```{r div_plot_together, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, ncol = 3))
```


### Mixed models

```{r mixed_rich_prepost, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>% 
    filter(!diet =="Wild")

Model_richness <- glmer.nb(richness ~ diet+(1|individual), data = alpha_div_meta)
summary(Model_richness)

Model_neutral <- lme(fixed = neutral ~ diet, data = alpha_div_meta,
               random = ~ 1 | individual)#log(seq_depth)+
summary(Model_neutral)

Model_phylo <- lme(fixed = phylogenetic ~ diet, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_phylo)

Model_funct <- lme(fixed = functional ~ diet, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_funct)
```


<!--chapter:end:05_alpha_diversity.Rmd-->

# Beta diversity

```{r load_data_beta1, comment="", message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")
load("data/beta.Rdata")
sample_metadata <- read_csv("data/sample_metadata.csv")
 
sample_metadata$individual <- sub("[0-9]{2}$", "", sample_metadata$animal)

sample_metadata_illu <- read_csv("data/metadata_illu.csv")
sample_metadata <- sample_metadata %>% 
  left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) 

samples_to_remove <- sample_metadata %>% 
  filter(individual == "GL_02_" | Samples %in% c("Blank", "Saliva") | group2 %in% c("Unknown") | season=="winter" ) %>% #!day %in% c("day_1", "Unknown")
  select(sample) %>% 
  pull()

beta_q1n_mat <- as.matrix(beta_q1n$S)
beta_q1n_filtered <- beta_q1n_mat[!rownames(beta_q1n_mat) %in% samples_to_remove,
                        !colnames(beta_q1n_mat) %in% samples_to_remove]
beta_q1n <- as.dist(beta_q1n_filtered)

sample_metadata <- sample_metadata %>% 
  filter(!sample %in% samples_to_remove)

genome_counts_filt <- genome_counts_filt %>% 
  select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome)

```


```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

genome_counts <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts_filt$genome)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, tree = genome_tree)
```

```{r save_beta, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
save(beta_q0n, 
     beta_q1n, 
     beta_q1p, 
     file = "data/beta.Rdata")
```

### Permanova analysis
```{r}
set.seed(1234)
```

**Richness**
```{r permanova_wildcap_q0, comment="", message=FALSE, warning=FALSE}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q0n$S), ]

betadisper(beta_q0n$S, sample_metadata_row$country) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$S ~ country, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

**Neutral**
```{r permanova_wildcap_q1, comment="", message=FALSE, warning=FALSE}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q1n), ]
sample_metadata_ordered <- sample_metadata_row %>% 
  rownames_to_column("sample")

betadisper(beta_q1n, sample_metadata_row$country) %>% permutest(., pairwise = TRUE) 

        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()

pairwise.adonis(beta_q1n, sample_metadata_ordered$country, perm = 999)

```

**Phylogenetic**
```{r permanova_wildcap_qP, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1p$S, sample_metadata_row$country) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1p$S ~ country, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

**Functional**
```{r permanova_wildcap_qF, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1f$S, sample_metadata_row$country) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1f$S ~ country,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

### Beta diversity plots

#### Richness

```{r beta_div_nmds_neutral_plot_median, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(country) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = country)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +labs(color='Origin')
```

#### Neutral

```{r beta_div_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}
beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(country) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = country)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +labs(color='Origin')
```


#### Phylogenetic

```{r beta_div_nmds_phylo1_plot, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(diet) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = diet)) +
    geom_point(size = 4) +
    scale_color_manual(values = diet_colors,labels=c("Pre_grass" = "Captive-bred no grass diet", "Wild" = "Wild"))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Origin')
```


#### Functional
```{r beta_div_nmds_func_plot, comment="", message=FALSE, warning=FALSE}
beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(diet) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = diet)) +
  geom_point(size = 4) +
  scale_color_manual(values = diet_colors,labels=c("Pre_grass" = "Captive-bred no grass diet", "Wild" = "Wild"))+
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme_classic() +
  theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Origin')
```


## Before and after grass

```{r load_data_prepost, comment="", message=FALSE, warning=FALSE}
load("data/beta_filtered_pre_post.Rdata")
```

```{r load_data_mag_prepost, comment="", message=FALSE, warning=FALSE}
genome_counts <- genome_counts_filt
post_pre_samples <- sample_metadata %>% 
  filter(diet!="Wild") %>%
  dplyr::select(sample) %>% pull()
genome_counts<- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(post_pre_samples))%>% 
  rownames_to_column("genome")
sample_metadata <- sample_metadata %>% 
  filter(diet!="Wild")
```
```{r beta_div_prepost, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

genome_counts <- genome_counts %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts$genome)

beta_q1p <- genome_counts %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, tree = genome_tree)

genome_counts_filt <- genome_counts[genome_counts$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  remove_rownames() %>% 
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

beta_q1f <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, dist = dist)

```


```{r save_beta_prepost, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
save(beta_q0n, 
     beta_q1n, 
     beta_q1p, 
     beta_q1f,
     genome_counts,
     sample_metadata,
     genome_metadata,
     genome_gifts,
     genome_gifts1,
     genome_tree,
     file = "data/beta_filtered_pre_post.Rdata")
```


### Permanova analysis
```{r set, comment="", message=FALSE, warning=FALSE}
set.seed(1234)
```

**Richness**
```{r permanova_pre_post_q0, comment="", message=FALSE, warning=FALSE}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q0n$S), ]

betadisper(beta_q0n$S, sample_metadata_row$diet) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$S ~ diet,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))),
        permutations = 999,
        strata = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

**Neutral**
```{r permanova_pre_post_q1, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$S, sample_metadata_row$diet) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$S ~ diet,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))),
        permutations = 999,
        strata = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

**Phylogenetic**
```{r permanova_pre_post_qP, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1p$S, sample_metadata_row$diet) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1p$S ~ diet,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))),
        permutations = 999,
        strata = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

**Functional**
```{r permanova_pre_post_qF, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1f$S, sample_metadata_row$diet) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1f$S ~ diet,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))),
        permutations = 999,
        strata = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

### Beta diversity plots

#### Richness

```{r beta_div_nmds_neutral_plot_medianpre_post, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(diet) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = diet)) +
    geom_point(size = 4) +
    scale_color_manual(values = diet_colors,labels=c("Pre_grass" = "Captive-bred no grass diet", "Post_grass" = "Captive-bred grass diet"))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Origin')
```

#### Neutral

```{r beta_div_neutral_pre_post, comment="", echo=FALSE, message=FALSE, warning=FALSE}
beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(diet) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = diet)) +
    geom_point(size = 4) +
    scale_color_manual(values = diet_colors,labels=c("Pre_grass" = "Captive-bred no grass diet", "Post_grass" = "Captive-bred grass diet"))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Origin')
```


#### Phylogenetic

```{r beta_div_nmds_phylo1_pre_post, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(diet) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = diet)) +
    geom_point(size = 4) +
    scale_color_manual(values = diet_colors,labels=c("Pre_grass" = "Captive-bred no grass diet", "Post_grass" = "Captive-bred grass diet"))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Origin')
```


#### Functional
```{r beta_div_nmds_func_pre_post, comment="", message=FALSE, warning=FALSE}
beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(diet) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = diet)) +
    geom_point(size = 4) +
    scale_color_manual(values = diet_colors,labels=c("Pre_grass" = "Captive-bred no grass diet", "Post_grass" = "Captive-bred grass diet"))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Origin')
```

<!--chapter:end:06_beta_diversity.Rmd-->

# AMR and VF

```{r load_data_amr, comment="", message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")
```

```{r load_genome_annotations, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_annotations <- read_tsv("data/gene_annotations.tsv.xz") %>% 
  mutate(genome = str_extract(gene, "^[^^]+")) %>% 
  mutate(ec = str_c("[EC:",ec,"]"))
```

## Genome size

```{r size_per_sample, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_genome_size <- genome_counts_filt %>% 
  mutate(across(-genome,~ .x / sum(.x, na.rm = TRUE))) %>% 
  inner_join(genome_metadata %>% select(genome,length), by = "genome") %>%
  pivot_longer(
    cols = -c(genome, length),
    names_to  = "sample",
    values_to = "rel_abundance"
  ) %>%
  group_by(sample) %>%
  summarise(
    size = weighted.mean(length, w = rel_abundance, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  left_join(sample_metadata,by="sample")
```

```{r size_per_sample_plot, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_genome_size %>% 
  ggplot(aes(x=country,y=size,color=country)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()

sample_genome_size %>% 
  ggplot(aes(x=breed,y=size,color=breed)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()

sample_genome_size %>% 
  ggplot(aes(x=locality,y=size,color=locality)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()
```

## AMR

```{r amr_per_genome, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_amr <- genome_annotations %>% 
  group_by(genome) %>%
  summarise(AMR = sum(resistance_type == "AMR", na.rm = TRUE), .groups = "drop")
```

```{r amr_per_sample, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_amr <- genome_counts_filt %>% 
  mutate(across(-genome,~ .x / sum(.x, na.rm = TRUE))) %>% 
  inner_join(genome_amr, by = "genome") %>%
  pivot_longer(
    cols = -c(genome, AMR),
    names_to  = "sample",
    values_to = "rel_abundance"
  ) %>%
  group_by(sample) %>%
  summarise(
    AMR = weighted.mean(AMR, w = rel_abundance, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  left_join(sample_metadata,by="sample")
```

```{r amr_per_sample_plot, warning=FALSE, comments="", message=FALSE, eval=FALSE}
amr_plot <- sample_amr %>% 
  ggplot(aes(x=group2,y=AMR,color=group2,fill=group2)) +
        geom_jitter(width = 0.3, show.legend = FALSE) +
      geom_boxplot(width = 0.7, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
      theme_classic() +
  theme(
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
    strip.background = element_blank(),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )
png(filename="amr_all.png", width = 680, height = 380,)
plot(amr_plot)
dev.off()

sample_amr %>% 
  ggplot(aes(x=breed,y=AMR,color=breed)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()

sample_amr %>% 
  ggplot(aes(x=locality,y=AMR,color=locality)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()
```

## Virulence factors

```{r vf_per_genome, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_vf <- genome_annotations %>% 
  group_by(genome) %>%
  summarise(vf = sum(!is.na(vf_type), na.rm = TRUE), .groups = "drop")
```

```{r vf_per_sample, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_vf <- genome_counts_filt %>% 
  mutate(across(-genome,~ .x / sum(.x, na.rm = TRUE))) %>% 
  inner_join(genome_vf, by = "genome") %>%
  pivot_longer(
    cols = -c(genome, vf),
    names_to  = "sample",
    values_to = "rel_abundance"
  ) %>%
  group_by(sample) %>%
  summarise(
    vf = weighted.mean(vf, w = rel_abundance, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  left_join(sample_metadata,by="sample")
```

```{r vf_per_sample_plot, warning=FALSE, comments="", message=FALSE, eval=FALSE}
vf_plot <- sample_vf %>% 
  ggplot(aes(x=group2,y=vf,color=group2,fill=group2)) +
        geom_jitter(width = 0.3, show.legend = FALSE) +
      geom_boxplot(width = 0.7, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
      theme_classic() +
  theme(
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
    strip.background = element_blank(),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )
png(filename="vf_all.png", width = 680, height = 380,)
plot(vf_plot)
dev.off()

sample_vf %>% 
  ggplot(aes(x=breed,y=vf,color=breed)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()

sample_vf %>% 
  ggplot(aes(x=locality,y=vf,color=locality)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter()
```

## Size-function correlations

```{r size_function_correlation, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_genome_size %>% 
  left_join(sample_amr %>% select(sample,AMR),by="sample") %>% 
  left_join(sample_vf %>% select(sample,vf),by="sample") %>% 
  left_join(alpha_div,by="sample") %>% 
  summarise(
    size_AMR = cor(size, AMR,  use = "complete.obs"),
    size_vf  = cor(size, vf,   use = "complete.obs"),
    size_richness  = cor(size, richness,   use = "complete.obs"),
    size_neutral  = cor(size, neutral,   use = "complete.obs"),
    size_phylogenetic  = cor(size, phylogenetic,   use = "complete.obs"),
    neutral_AMR = cor(neutral, AMR,   use = "complete.obs")
  )
```

```{r size_amr_correlation_plot, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_genome_size %>% 
  left_join(sample_amr %>% select(sample,AMR),by="sample") %>% 
  left_join(sample_vf %>% select(sample,vf),by="sample") %>% 
  left_join(alpha_div,by="sample") %>% 
  ggplot(aes(x=size,y=AMR, color=country)) + 
    geom_point() +
    geom_smooth(method = "lm", se = TRUE)
```

```{r size_vf_correlation_plot, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_genome_size %>% 
  left_join(sample_amr %>% select(sample,AMR),by="sample") %>% 
  left_join(sample_vf %>% select(sample,vf),by="sample") %>% 
  left_join(alpha_div,by="sample") %>% 
  ggplot(aes(x=size,y=vf, color=country)) + 
    geom_point() +
    geom_smooth(method = "lm", se = TRUE)
```


<!--chapter:end:08_AMR_VF.Rmd-->

# Greenlandic sled dogs in summer

```{r knitr_options_beta, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(24022025)
```

## Filter data

```{r load_data_G_community, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

```{r load_data_host_filt, comment="", message=FALSE, warning=FALSE}
sample_metadata <- read_csv("data/sample_metadata.csv")

 sample_metadata <- sample_metadata %>%
   filter(country== "Greenland")

# sample_metadata <- sample_metadata %>%
#   filter(country== "Greenland" | breed == "Wolf")

sample_metadata$individual <- sub("[0-9]{2}$", "", sample_metadata$animal)

sample_metadata_illu <- read_csv("data/metadata_illu.csv")
sample_metadata <- sample_metadata %>% 
  left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) %>% 
  filter(!Samples %in% c("Blank","Saliva")) %>% 
  filter(individual!="GL_02_") %>% 
  filter(season!="winter")

genome_counts_filt <- genome_counts_filt %>% 
    select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips
```


## Taxonomy overview 

**Number of bacteria phyla**
```{r phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**After merging all Bacillota**
```{r}
genome_metadata <- genome_metadata %>%
  mutate(phylum_clean = ifelse(grepl("^Bacillota_", phylum), "Bacillota", phylum))

genome_metadata %>% 
  filter(domain == "Bacteria")%>%
  dplyr::select(phylum_clean) %>%
  unique() %>%
  pull() %>%
  length()
```

### Stacked barplot

```{r taxonomy_G_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),  ~ . / sum(.)) %>%
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(., genome_metadata, by = join_by(genome == genome)) %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(count > 0) %>%
  ggplot(., aes(x = sample,
    y = count,
    fill = phylum,
    group = phylum
  )) +
  geom_bar(stat = "identity",
           colour = "white",
           linewidth = 0.1) +
  scale_fill_manual(values = phylum_colors) +
  facet_grid( ~ locality, scale = "free", space = "free") +
  guides(fill = guide_legend(ncol = 1)) +
  theme(
    axis.title.x = element_blank(),
    panel.background = element_blank(),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    strip.background = element_rect(fill = "white"),
    strip.text = element_text(size = 8, lineheight = 0.6),
    strip.placement = "outside",
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.line = element_line(
      linewidth = 0.5,
      linetype = "solid",
      colour = "black"
    )
  ) +
  labs(fill = "Phylum", y = "Relative abundance", x = "Samples")
```

### Phylum relative abundances

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum_clean,locality) %>%
  summarise(relabun=sum(count))
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum_clean) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    select(phylum_clean) %>%
    pull()
```
```{r taxonomy_phylum_summary_origin, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
  group_by(phylum_clean) %>%
  summarise(
    total_mean = mean(relabun * 100, na.rm = T),
    total_sd = sd(relabun * 100, na.rm = T),
    Daneborg_mean = mean(relabun[locality == "Daneborg"] * 100, na.rm =T),
    Daneborg_sd = sd(relabun[locality == "Daneborg"] * 100, na.rm =T),
    Disko_mean = mean(relabun[locality == "Disko"] * 100, na.rm = T),
    Disko_sd = sd(relabun[locality == "Disko"] * 100, na.rm =T),
    Illulissat_mean = mean(relabun[locality == "Illulissat"] * 100, na.rm = T),
    Illulissat_sd = sd(relabun[locality == "Illulissat"] * 100, na.rm = T),
    Ittoqqortoormiit_mean = mean(relabun[locality == "Ittoqqortoormiit"] * 100, na.rm = T),
    Ittoqqortoormiit_sd = sd(relabun[locality == "Ittoqqortoormiit"] * 100, na.rm = T),
    Wolf_mean = mean(relabun[locality == "Mongolia"] * 100, na.rm = T),
    Wolf_sd = sd(relabun[locality == "Mongolia"] * 100, na.rm = T)) %>%
  mutate(
    total = str_c(round(total_mean, 4), "±", round(total_sd, 3)),
    Daneborg = str_c(round(Daneborg_mean, 4), "±", round(Daneborg_sd, 3)),
    Disko = str_c(round(Disko_mean, 4), "±", round(Disko_sd, 3)),
    Illulissat = str_c(round(Illulissat_mean, 4), "±", round(Illulissat_sd, 3)),
    Ittoqqortoormiit = str_c(round(Ittoqqortoormiit_mean, 4), "±", round(Ittoqqortoormiit_sd, 3)),
    Wolf = str_c(round(Wolf_mean, 4), "±", round(Wolf_sd, 3))
  ) %>%
  arrange(-total_mean) %>%
  dplyr::select(phylum_clean, total, Daneborg, Disko,Illulissat,Ittoqqortoormiit, Wolf)
```



```{r alpha_G_div, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")


genome_counts_filt_beta <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt_beta <- genome_counts_filt_beta %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt_beta$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  filter(genome %in% rownames(dist)) %>% 
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))%>%
  full_join(functional, by = join_by(sample == sample))
```

### Plots

```{r alpha_div_G_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Functional")
```

```{r div_plot_G, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3,plot4, ncol = 2))
```

### Values

```{r alpha_values_season, comment="", message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(alpha) %>%
  summarise(
    total_mean = mean(value, na.rm = T),
    total_sd = sd(value, na.rm = T),
    Daneborg_mean = mean(value[locality == "Daneborg"], na.rm =T),
    Daneborg_sd = sd(value[locality == "Daneborg"], na.rm =T),
    Disko_mean = mean(value[locality == "Disko"], na.rm = T),
    Disko_sd = sd(value[locality == "Disko"], na.rm =T),
    Illulissat_mean = mean(value[locality == "Illulissat"], na.rm = T),
    Illulissat_sd = sd(value[locality == "Illulissat"], na.rm = T),
    Ittoqqortoormiit_mean = mean(value[locality == "Ittoqqortoormiit"], na.rm = T),
    Ittoqqortoormiit_sd = sd(value[locality == "Ittoqqortoormiit"], na.rm = T),,
    Wolf_mean = mean(value[locality == "Mongolia"], na.rm = T),
    Wolf_sd = sd(value[locality == "Mongolia"], na.rm = T)) %>%
  mutate(
    total = str_c(round(total_mean, 3), "±", round(total_sd, 3)),
    Daneborg = str_c(round(Daneborg_mean, 3), "±", round(Daneborg_sd, 3)),
    Disko = str_c(round(Disko_mean, 3), "±", round(Disko_sd, 3)),
    Illulissat = str_c(round(Illulissat_mean, 3), "±", round(Illulissat_sd, 3)),
    Ittoqqortoormiit = str_c(round(Ittoqqortoormiit_mean, 3), "±", round(Ittoqqortoormiit_sd, 3)),
    Wolf = str_c(round(Wolf_mean, 3), "±", round(Wolf_sd, 3))
  ) %>%
  arrange(-total_mean) %>%
  dplyr::select(alpha,total, Daneborg, Disko,Illulissat,Ittoqqortoormiit, Wolf)

```

### Mixed models

```{r mixed_rich_prepost, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))

Model_richness <- MASS::glm.nb(richness ~ locality, data = alpha_div_meta,trace=TRUE)
anova(Model_richness)
summary(Model_richness)
emmeans(Model_richness, pairwise ~ locality)

Model_neutral<- lm(formula = neutral ~ locality, data = alpha_div_meta) 
anova(Model_neutral)
summary(Model_neutral)
emmeans(Model_neutral, pairwise ~ locality)

Model_phylo<- lm(formula = phylogenetic ~ locality, data = alpha_div_meta) 
anova(Model_phylo)
summary(Model_phylo)
emmeans(Model_phylo, pairwise ~ locality)

Model_functional<- lm(formula = functional ~ locality, data = alpha_div_meta) 
anova(Model_functional)
summary(Model_functional)
emmeans(Model_functional, pairwise ~ locality)
```

```{r beta_div_G, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

genome_counts <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts_filt$genome)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, tree = genome_tree)

genome_counts_filt_beta <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt_beta <- genome_counts_filt_beta %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt_beta$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

beta_q1f <- genome_counts_filt_beta %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, dist = dist)
```

```{r save_beta_G, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
save(beta_q0n, 
     beta_q1n, 
     beta_q1p,
     beta_q1f,
     file = "data/beta_greenland_wolf.Rdata")
```

### Beta diversity plots

```{r load_data_beta_summer, comment="", message=FALSE, warning=FALSE}
load("data/beta_greenland_summer.Rdata")
```

#### Richness

```{r beta_div_nmds_neutral_plot_G, comment="", message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +labs(color='Origin')
```

#### Neutral

```{r beta_div_neutral_G, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=20, fig.fullwidth=TRUE}
beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +
  labs(color='Locality')
# +
#   geom_text_repel(aes(label = animal), size=3)

```


#### Phylogenetic

```{r beta_div_nmds_phylo1_plot_G, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Locality')
```

#### Functional
```{r beta_div_nmds_func_plot_G, comment="", message=FALSE, warning=FALSE}
beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
  geom_point(size = 4) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme_classic() +
  theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Locality')
```

### Permanova analysis
```{r}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q1n$S), ]
```

**Richness**
```{r permanova_locality_q0, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q0n$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$S ~ locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```

**Neutral**
```{r permanova_locality_q1, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$S ~ locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_neutral_locality, comment="", message=FALSE, warning=FALSE}
sample_metadata <- sample_metadata %>% arrange(match(sample,labels(beta_q1n$S)))
pairwise.adonis(beta_q1n$S, sample_metadata$locality, perm = 999)
```

**Phylogenetic**
```{r permanova_locality_qP, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1p$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1p$S ~ locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_phylo, comment="", message=FALSE, warning=FALSE}
sample_metadata <- sample_metadata %>% arrange(match(sample,labels(beta_q1p$S)))

pairwise.adonis(beta_q1p$S, sample_metadata$locality, perm = 999)
```

**Functional**
```{r permanova_locality_qF, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1f$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1f$S ~ place*locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_functional, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(beta_q1f$S, sample_metadata$locality, perm = 999)
```



## Enrichment analysis: Ancombc2

```{r zero_phylo1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
phylo_samples <- sample_metadata %>%
  column_to_rownames("sample") %>%
  sample_data()
phylo_genome <- genome_counts_filt %>%
  select(one_of(c("genome", rownames(phylo_samples)))) %>%
  column_to_rownames("genome") %>%
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>%
  filter(genome %in% rownames(phylo_genome)) %>%
  mutate(genome2 = genome) %>% 
  column_to_rownames("genome2") %>%
  dplyr::select(domain, phylum, class, order, family, genus, species, genome) %>% 
  as.matrix() %>%
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r save_ancombc_summer, warning=FALSE, comments="", message=FALSE}
load("data/ancombc_results.Rdata") #ancombc_results_wolves.Rdata
```

### MAG level
```{r ancom_rand_mag, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_mag = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = NULL,
  fix_formula = "locality",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "locality",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

### the global differential taxa
```{r}
res_global <- ancom_rand_output_mag$res_global
global_hits <- res_global %>%
  filter(diff_abn == TRUE) %>%
  select(taxon, W, p_val, q_val)
```

### Extract the results
```{r}
 res_pair <- ancom_rand_output_mag$res_pair
```

```{r labeling, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}

# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


```

```{r}
plot_data <- plot_data %>%
  left_join(genome_metadata %>% select(genome, phylum), by =join_by("taxon"=="genome"))

```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by = join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
  unique() %>%
  dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])#"Comparison:", 
  
  subset_data <- plot_data %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data$phylum)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data, aes(x = taxon_ordered, y = lfc, fill = phylum)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 6),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14, face = "bold"),
      legend.position = "right",
      legend.box = "vertical",
      plot.title = element_text(size = 18, face = "bold", hjust = 0.5)
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      #paste("Enriched Taxa:", c)
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Phylum"
    )
  
  ggsave(paste0("Fig_", gsub(" ", "_", c), ".png"), plot = p, width = 10, height = 25, dpi = 300)
}
```

# top 10
```{r}
for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])

  subset_data <- plot_data %>%
    group_by(contrast) %>%
    filter(lfc > 0) %>%
    slice_max(order_by = lfc, n = 10) %>%
    bind_rows(
      plot_data %>%
        group_by(contrast) %>%
        filter(lfc < 0) %>%
        slice_min(order_by = lfc, n = 10)
    ) %>%
    ungroup() %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data$phylum)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data, aes(x = taxon_ordered, y = lfc, fill = phylum)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14, face = "bold"),
      legend.position = "right",
      legend.box = "vertical",
      plot.title = element_text(size = 18, face = "bold")
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Phylum"
    )
  
  ggsave(paste0("Fig_top10_", gsub(" ", "_", c), ".png"), plot = p, width = 8, height = 12, dpi = 300)
}
```

### Phylum level
```{r ancom_rand_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_phylum = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = "phylum",
  fix_formula = "locality",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "locality",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r}
res_pair_phylum <- ancom_rand_output_phylum$res_pair
```

```{r  labeling_phylum, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair_phylum), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data_phylum <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_phylum %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )
```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data_phylum$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)


for (c in unique_contrasts) {
  
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])# 

  
  subset_data_phylum <- plot_data_phylum %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data_phylum$taxon)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data_phylum, aes(x = taxon_ordered, y = lfc, fill = taxon)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(
        size = 0.5,
        linetype = "solid",
        colour = "black"
      ),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_blank(),
      legend.title = element_blank(),
      plot.title = element_text(size = 18, face = "bold", hjust = 0.5)
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      x = "Taxa",
      y = "Log-Fold Change",
    )
  
  ggsave(paste0("Fig_phylum_", gsub(" ", "_", c), ".png"), plot = p, width = 6, height = 8, dpi = 300)
}
```

### Genus level
```{r ancom_rand_genus, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_genus = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = "genus",
  fix_formula = "locality",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "locality",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r save_ancombc, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(ancom_rand_output_mag,
     ancom_rand_output_phylum,
     ancom_rand_output_genus,
     file = "data/ancombc_results.Rdata") #ancombc_results_wolves.Rdata
```


```{r}
res_pair_genus <- ancom_rand_output_genus$res_pair
```


```{r labeling_genus, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}

# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair_genus), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data_genus <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_genus %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


```

```{r}
plot_data_genus <- plot_data_genus %>%
  mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
  left_join(., genome_metadata %>% select(genus, family, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(phylum.x = coalesce(phylum.x, phylum.y)) %>% 
  select(-family,-phylum.y) %>% 
  rename(phylum=phylum.x)

```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data_genus$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  titulu <- paste(locs[2], " - ", locs[1])
  
  subset_data_genus <- plot_data_genus %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) 
#  %>%  mutate(taxon_ordered = factor(taxon_clean, levels = unique(taxon_clean)))
  
  tax_color <- as.data.frame(unique(subset_data_genus$phylum)) %>% 
    rename(phylum=1) %>% 
    left_join(., colors_alphabetic, by="phylum")%>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()

  
  p <- ggplot(subset_data_genus, aes(x=lfc, y=forcats::fct_reorder(taxon_clean,lfc), fill=phylum))  +
    geom_bar(stat = "identity") +
 #   coord_flip() +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical",
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))+   
      scale_fill_manual(values=tax_color) + 
    labs(
      title = titulu,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Direction"
    )
  
  ggsave(paste0("Fig_genus_", gsub(" ", "_", c), ".png"), plot = p, width = 6, height = 10, dpi = 300)
}
```

```{r}
# Loop through each contrast
plot_data_genus <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_genus %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


plot_data_genusF <- plot_data_genus %>%
  mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
  left_join(., genome_metadata %>% select(genus, class, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, class, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(class.x = coalesce(class.x, class.y)) %>% 
  select(-class.y) %>% 
  rename(class=class.x)

unique_contrasts <- unique(plot_data_genus$contrast)

# taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
#   rownames_to_column(., "taxon")

# colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#   mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
#   right_join(taxonomy, by=join_by(phylum == phylum)) %>%
#   dplyr::select(phylum, colors) %>%
#   mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
# 	unique() %>%
# 	dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  titulu <- paste(locs[2], " - ", locs[1])
  
  subset_data_genusF <- plot_data_genusF %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) 
#  %>%  mutate(taxon_ordered = factor(taxon_clean, levels = unique(taxon_clean)))
  
  # tax_color <- as.data.frame(unique(subset_data_genus$phylum)) %>% 
  #   rename(phylum=1) %>% 
  #   left_join(., colors_alphabetic, by="phylum")%>%
  #   dplyr::arrange(phylum) %>%
  #   dplyr::select(colors) %>%
  #   pull()

  
  p_f <- ggplot(subset_data_genusF, aes(x=lfc, y=forcats::fct_reorder(taxon_clean,lfc), fill=class))  +
    geom_bar(stat = "identity") +
 #   coord_flip() +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical",
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))+
    scale_fill_discrete()+
    labs(
      title = titulu,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Faily"
    )
  
  ggsave(paste0("Fig_genus_color_family_", gsub(" ", "_", c), ".png"), plot = p_f, width = 10, height = 20, dpi = 300)
}
```

## Functional differences

```{r gift_analysis, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>% 
  select(genome, all_of(sample_metadata$sample)) %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

#Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

#Get community-weighed average GIFTs per sample
genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 
# genome_counts_row <- rownames_to_column(genome_counts_row, "genome")
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

## Groups MCI
```{r gitfs_functional_wild, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  summarise(MCI = mean(value), sd = sd(value)) %>%
  tt()
```

```{r}
MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
kruskal.test(value ~ locality, data=MCI)
```

## Community elements differences:
```{r comunity_elem_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,7)], by="sample")
```

```{r commun_wilcox_elem_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,locality), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ locality)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

selected_data <- element_gift %>%
  pivot_longer(-c(sample, locality), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_elements$trait)

# Now run Dunn's test per trait
library(rstatix)

pairwise_results <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ locality, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element))
```
```{r}
pairwise_results %>% group_by(comparison) %>%
  summarise(num_significant_functions = n()) %>% 
  arrange(num_significant_functions)
```
```{r function_plot_summer, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Locality Comparison",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )
```
```{r barplot_func, comment="", message=FALSE, warning=FALSE, fig.height=60, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,5,7)], by="sample")

plot_data <- element_gift %>%
  pivot_longer(-c(sample, locality, origin), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results$trait)%>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element))

# Plot boxplots per trait
ggplot(plot_data, aes(x = locality, y = value, fill = locality)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(aes(shape = origin), width = 0.2, size = 3) +
  #geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```
```{r}
plot_data %>% group_by(locality, Function) %>% summarise(mean=mean(value)) %>% pivot_wider(names_from = locality, values_from = c(mean))
```
```{r}
plot_data %>% group_by(locality, Function) %>% summarise(mean=mean(value)) %>% pivot_wider(names_from = locality, values_from = c(mean)) %>% select(Function) %>% pull()
```

## Community function differences:
```{r comunity_func_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,7)], by="sample")
```

```{r commun_wilcox_func_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueFuncGIFT_db<- unique(GIFT_db[c(3,4,5)])

significant_functions <- function_gift %>%
    pivot_longer(-c(sample,locality), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ locality)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "functions")  %>% 
  select(-functions)

selected_data <- function_gift %>%
  pivot_longer(-c(sample, locality), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_functions$trait)

# Now run Dunn's test per trait
library(rstatix)

pairwise_results_func <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ locality, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function))

```

```{r pairwise_plot_func, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results_func, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Locality Comparison",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )

```

```{r barplot_func_wolves, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,5,7)], by="sample")

plot_data <- function_gift %>%
  pivot_longer(-c(sample, locality, origin), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results_func$trait)%>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function)) 
# %>% 
#   filter(Function=="Xenobiotic degradation_Mercury")

# Plot boxplots per trait
ggplot(plot_data, aes(x = locality, y = value, fill = locality)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(aes(shape = origin), width = 0.2, size = 3) +
  #geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```




<!--chapter:end:greenland.Rmd-->

# Winter vs summer in eastern Greenland

```{r load_data_host_filt_season, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")

sample_metadata <- read_csv("data/sample_metadata.csv")
 
sample_metadata$individual <- sub("[0-9]{2}$", "", sample_metadata$animal)

sample_metadata_illu <- read_csv("data/metadata_illu.csv")
sample_metadata <- sample_metadata %>% 
  left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) 

# sample_metadata <- sample_metadata%>%
#   filter(country== "Greenland")

load("data/beta.Rdata")

samples_to_keep <- sample_metadata %>% 
  filter(locality == "Ittoqqortoormiit") %>% 
#  filter(group2=="Wolf" | locality%in% c("Ittoqqortoormiit", "Daneborg")) %>%
  filter(day %in% c("day_1", "Unknown")) %>% 
# filter(individual != "SD03126") %>% 
  # filter(!Samples %in% c("Blank", "Saliva"))%>% 
  select(sample) %>% 
  pull()

beta_q1n_mat <- as.matrix(beta_q1n$S)
beta_q1n_filtered <- beta_q1n_mat[rownames(beta_q1n_mat) %in% samples_to_keep,
                        colnames(beta_q1n_mat) %in% samples_to_keep]
beta_q1n <- as.dist(beta_q1n_filtered)

sample_metadata <- sample_metadata %>% 
  filter(sample %in% samples_to_keep)

genome_counts_filt <- genome_counts_filt %>% 
  select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome)


# sample_metadata <- sample_metadata %>% 
#   left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) %>% 
#   filter(!Samples %in% c("Blank","Saliva")) %>% 
#   filter(individual!="GL_02_") %>% 
#   filter(group2 %in% c("Wolf", "Sled_dogs")) %>% #, "Daneborg"
# #  filter(season=="winter") %>% 
#   filter(day %in% c("day_1", "Unknown"))

genome_counts_filt <- genome_counts_filt %>% 
    select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome)

```

```{r alpha_div_season, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))
```


```{r alpha_div_boxplot_season, comment="",echo=FALSE, message=FALSE, warning=FALSE}
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = season, group=season, color=season, fill=season)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = season, group=season, color=season, fill=season)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = season, group=season, color=season, fill=season)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Phylogenetic")
```

```{r div_plot_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, ncol = 3))
```

```{r beta_div_season, comment="", message=FALSE, warning=FALSE}
beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

```


### Beta diversity plots


#### Neutral

```{r beta_div_neutral_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1n %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(group1) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = group1)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +
  labs(color='Group') +
   geom_text_repel(aes(label = individual), size=3)

```

### Permanova analysis
```{r}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q1n), ]
```

**Neutral**
```{r permanova_season_q1n, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n, sample_metadata_row$group1 %>% arrange(match(rownames,labels(beta_q1n)))) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n ~ group1, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_neutral_locality, comment="", message=FALSE, warning=FALSE}
# sample_metadata <- sample_metadata %>% arrange(match(sample,labels(beta_q1n)))
# pairwise.adonis(beta_q1n, sample_metadata$group1, perm = 999)
```


## Enrichment analysis: Ancombc2

```{r zero_phylo1_season, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>%
  column_to_rownames("sample") %>%
  sample_data()
phylo_genome <- genome_counts_filt %>%
  select(one_of(c("genome", rownames(phylo_samples)))) %>%
  column_to_rownames("genome") %>%
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>%
  filter(genome %in% rownames(phylo_genome)) %>% 
  mutate(genome2 = genome) %>% 
  column_to_rownames("genome2") %>%
  dplyr::select(domain, phylum, class, order, family, genus, species, genome) %>% 
  as.matrix() %>%
  tax_table() 

physeq_genome_itto <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_itto <- prune_taxa(taxa_sums(physeq_genome_itto)>0, physeq_genome_itto)

```

```{r load_ancombc_season, warning=FALSE, comments="", message=FALSE}
load("data/ancombc_results_itto.Rdata") #ancombc_results_wolves.Rdata
```

#### MAG level
```{r ancom_rand_mag_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_mag_itto = ancombc2(
  data = physeq_genome_itto,
  assay_name = "counts",
  tax_level = NULL,
  fix_formula = "season",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "season",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = FALSE,
  pairwise = FALSE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

#### Structural zeros

***Total number of structural zeros***
```{r}
# Extract structural zeros
structural_zeros <- ancom_rand_output_mag_itto$zero_ind

# Convert to a data frame for better readability
structural_zeros_df <- as.data.frame(structural_zeros) %>% 
  column_to_rownames(., "taxon")

structural_zero_taxa <- rownames(structural_zeros_df)[apply(structural_zeros_df, 1, any)]

# Identify in which groups each taxon is a structural zero
structural_zero_groups <- structural_zeros_df[structural_zero_taxa, , drop = FALSE] %>% 
  rename("summer"=1) %>% 
  rename("winter"=2)

# Total number of structural zeros
nrow(structural_zero_groups)

```

***Structural zeros in summer***
```{r}
structural_zero_groups %>% 
  filter(summer==TRUE) %>% 
  nrow()

structural_zero_groups %>% 
  filter(summer==TRUE) %>% 
  rownames_to_column(., "genome") %>% 
  left_join(., genome_metadata, by="genome") %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 
```

***Structural zeros in winter***
```{r}
structural_zero_groups %>% 
  filter(winter==TRUE) %>% 
  nrow()

structural_zero_groups %>% 
  filter(winter==TRUE) %>% 
  rownames_to_column(., "genome") %>% 
  left_join(., genome_metadata, by="genome") %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 
```

***Structural zeros in wolves***
```{r}
structural_zero_groups %>% 
  filter(Wolf==TRUE) %>% 
  nrow()

structural_zero_groups %>% 
  filter(Wolf==TRUE) %>% 
  rownames_to_column(., "genome") %>% 
  left_join(., genome_metadata, by="genome") %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 
```

```{r ancom_table_season, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_itto@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(phylum, order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag_itto <- ancom_rand_output_mag_itto$res %>%
  dplyr::select(taxon, lfc_seasonwinter, q_seasonwinter) %>%
  filter(q_seasonwinter < 0.05) %>%
  dplyr::arrange(q_seasonwinter) %>%
  left_join(taxonomy, by = join_by(taxon == taxon)) %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_seasonwinter)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_itto$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag_itto%>%
      mutate(genome=factor(genome,levels=ancombc_rand_table_mag_itto$genome)) %>%
ggplot(., aes(x=lfc_seasonwinter, y=forcats::fct_reorder(genome,lfc_seasonwinter), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 6),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  geom_text(aes(4, 20), label = "Enriched\nin winter", color="#666666") +
  geom_text(aes(-3.5, 15), label = "Enriched\nin summer", color="#666666")+
  xlab("log2FoldChange") + 
  ylab("Genome")+
  guides(fill=guide_legend(title="Phylum"))
```

### Phylum level
```{r ancom_rand_phylum_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_phylum_itto = ancombc2(
  data = physeq_genome_itto,
  assay_name = "counts",
  tax_level = "phylum",
  fix_formula = "season",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "season",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r}
res_pair_phylum_itto <- ancom_rand_output_phylum_itto$res_pair

ancom_rand_output_phylum_itto$res %>%
  dplyr::select(taxon, lfc_seasonwinter, q_seasonwinter) %>%
  filter(q_seasonwinter < 0.05) %>%
  dplyr::arrange(q_seasonwinter)

```


### Genus level
```{r ancom_rand_genus_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_genus_itto = ancombc2(
  data = physeq_genome_itto,
  assay_name = "counts",
  tax_level = "genus",
  fix_formula = "season",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "season",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r}
# ancom_rand_output_genus_itto$res %>%
#   dplyr::select(taxon, lfc_seasonwinter, q_seasonwinter) %>%
#   filter(q_seasonwinter < 0.05) %>%
#   dplyr::arrange(q_seasonwinter)%>%
#   mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
#   left_join(., genome_metadata %>% select(genus, family, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
#   left_join(., genome_metadata %>% select(family, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
#   mutate(phylum.x = coalesce(phylum.x, phylum.y)) %>% 
#   select(-family,-phylum.y) %>% 
#   rename(phylum=phylum.x)

taxonomy <- data.frame(physeq_genome_itto@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

ancombc_rand_table_genus_itto <- ancom_rand_output_genus_itto$res %>%
  dplyr::select(taxon, lfc_seasonwinter, q_seasonwinter) %>%
  filter(q_seasonwinter < 0.05) %>%
  dplyr::arrange(q_seasonwinter)%>%
  mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
  left_join(., genome_metadata %>% select(genus, family, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(phylum.x = coalesce(phylum.x, phylum.y)) %>% 
  select(-family,-phylum.y) %>% 
  rename(phylum=phylum.x)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_genus_itto$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```
```{r}
ancombc_rand_table_mag_itto%>%
      mutate(genome=factor(genome,levels=ancombc_rand_table_mag_itto$genome)) %>%
ggplot(., aes(x=lfc_seasonwinter, y=forcats::fct_reorder(genome,lfc_seasonwinter), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 6),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Genome")+
  guides(fill=guide_legend(title="Phylum"))


ggplot(ancombc_rand_table_genus_itto, aes(x=lfc_seasonwinter, y=forcats::fct_reorder(taxon_clean,lfc_seasonwinter), fill=phylum))  +
    geom_bar(stat = "identity") +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 6),
        axis.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical",
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))+   
      scale_fill_manual(values=tax_color) + 
    labs(
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Direction"
    )
```



```{r save_ancombc_season, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(ancom_rand_output_mag_itto,
     ancom_rand_output_phylum_itto,
     ancom_rand_output_genus_itto,
     file = "data/ancombc_results_itto.Rdata") #ancombc_results_wolves.Rdata
```

```{r gift_analysis_season, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

#Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

#Get community-weighed average GIFTs per sample
genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 
# genome_counts_row <- rownames_to_column(genome_counts_row, "genome")
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

## Groups MCI
```{r gitfs_functional_season, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(season) %>%
  summarise(MCI = mean(value), sd = sd(value)) %>%
  tt()
  
```

***Shapiro test***
```{r}
MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample))

shapiro.test(MCI$value)
wilcox.test(value ~ season, data=MCI)
```

### Community elements differences:
```{r comunity_elem_season, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(1,8)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_elem_season, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,season), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ season)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,8)], by = join_by(sample == sample))

element_gift_filt %>%
  select(-sample)%>%
  group_by(season)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(season) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=summer-winter)%>% 
  mutate(group_color = ifelse(Difference <0, "winter","summer")) 
```

```{r plot3_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function2 <- difference_table %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes<-unique(code_function2$Code_function)

gift_colors <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes)%>% 
  mutate(legend=str_c(Code_function," - ",Function))
```

```{r final_plot_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function2 %>%
  left_join(gift_colors[c(1,3,4)], by=join_by(Code_function==Code_function))%>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=legend)) + 
  geom_col() +
  scale_color_manual(values = gift_colors$Color)+
  geom_hline(yintercept=0) + 
  coord_flip()+
  geom_text(aes(2, 0.028), label = "Enriched in\n summer", color="#666666") +
  geom_text(aes(1.7, -0.025), label = "Enriched in\n winter ", color="#666666") +
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank())+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  labs(fill="Season")
```

```{r}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,12)], by="sample")

uniqueFuncGIFT_db<- unique(GIFT_db[c(3,4,5)])

significant_functions <- function_gift %>%
    pivot_longer(-c(sample,group1), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ group1)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "functions")  %>% 
  select(-functions) %>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function))


function_gift_t <- function_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

function_gift_filt <- subset(function_gift_t, trait %in% significant_functions$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,8)], by = join_by(sample == sample))

function_gift_filt %>%
  select(-sample)%>%
  group_by(season)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")  %>%
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function))

difference_table <- function_gift_filt %>%
  select(-sample) %>%
  group_by(season) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "trait")  %>%
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function)) %>% 
  mutate(Difference=summer-winter)%>% 
  mutate(group_color = ifelse(Difference <0, "winter","summer")) 

  ggplot(difference_table, aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=legend)) + 
  geom_col() +
  scale_color_manual(values = gift_colors$Color)+
  geom_hline(yintercept=0) + 
  coord_flip()+
  geom_text(aes(2, 0.028), label = "Enriched in\n summer", color="#666666") +
  geom_text(aes(1.7, -0.025), label = "Enriched in\n winter ", color="#666666") +
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank())+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  labs(fill="Season")



selected_data <- function_gift %>%
  pivot_longer(-c(sample, group1), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_functions$trait)%>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function))
```
```{r}
selected_data %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=legend)) + 
  geom_col() +
  scale_color_manual(values = gift_colors$Color)+
  geom_hline(yintercept=0) + 
  coord_flip()+
  geom_text(aes(2, 0.028), label = "Enriched in\n summer", color="#666666") +
  geom_text(aes(1.7, -0.025), label = "Enriched in\n winter ", color="#666666") +
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank())+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  labs(fill="Season")
```

# bacteria differences
```{r}
ecoli <- genome_metadata %>%
  filter(species=="Escherichia coli") %>%
  select(genome) %>%
  pull()

ecoli_counts <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% ecoli)

ecoli_means <- ecoli_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)%>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata[c(1, 12)], by = join_by(sample == sample))

shapiro.test(ecoli_means$value)
wilcox.test(value ~ group1, data=ecoli_means)

ecoli_means <- ecoli_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)%>%
  rownames_to_column(., "sample") %>% 
  mutate(
    row_mean = rowMeans(select(., -sample), na.rm = TRUE),
    row_median = apply(select(., -sample), 1, median, na.rm = TRUE)
  ) %>% 
  select(sample, row_mean, row_median) %>%
  left_join(sample_metadata[c(1, 12)], by = join_by(sample == sample))

shapiro.test(ecoli_means$row_mean)
wilcox.test(row_mean ~ group1, data=ecoli_means)

shapiro.test(ecoli_means$row_median)
wilcox.test(row_median ~ group1, data=ecoli_means)
```

```{r plot3_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ecoli_counts%>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  mutate(row_median = apply(select(., -sample), 1, median, na.rm = TRUE)) %>% 
  select(sample, row_median) %>% 
  left_join(., sample_metadata[c(1,12)], by = join_by(sample == sample)) %>% 
  ggplot(., aes(x = group1, y = row_median, fill = group1)) +
  geom_violin(outlier.shape = NA, alpha = 0.7) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "E. coli Distributions Across Groups",
    x = "Groups",
    y = "E. coli" )
```

```{r plot3_season, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ecoli_counts%>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  mutate(row_mean = rowMeans(select(., -sample), na.rm = TRUE)) %>% 
  select(sample, row_mean) %>% 
  left_join(., sample_metadata[c(1,12)], by = join_by(sample == sample))%>% 
  ggplot(., aes(x = group1, y = row_mean, fill = group1)) +
  geom_violin(alpha = 0.7) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```

```{r}
bacteria <- genome_metadata %>%  filter(species=="Campylobacter_D jejuni") %>%  select(genome) %>%  pull()

bacteria_counts <- genome_counts_filt %>% filter(genome %in% bacteria)


bacteria_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,12)], by = join_by(sample == sample))%>%
  group_by(group1) %>% 
  summarise(mean=mean(SRR14842419_bin_269, na.rm = TRUE))
```
```{r}
bacteria <- genome_metadata %>%
  filter(species=="Klebsiella pneumoniae") %>%
  select(genome) %>%
  pull()

bacteria_counts <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% bacteria)

bacteria_means <- bacteria_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)%>%
  rownames_to_column(., "sample") %>% 
  mutate(
    row_mean = rowMeans(select(., -sample), na.rm = TRUE),
    row_median = apply(select(., -sample), 1, median, na.rm = TRUE)
  ) %>% 
  select(sample, row_mean, row_median) %>%
  left_join(sample_metadata[c(1, 12)], by = join_by(sample == sample))

shapiro.test(bacteria_means$row_mean)
wilcox.test(row_mean ~ group1, data=bacteria_means)

shapiro.test(bacteria_means$row_median)
wilcox.test(row_median ~ group1, data=bacteria_means)
```

```{r}
bacteria <- genome_metadata %>%  filter(species=="Klebsiella pneumoniae") %>%  select(genome) %>%  pull()

bacteria_counts <- genome_counts_filt %>% filter(genome %in% bacteria)

bacteria_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  mutate(row_mean = rowMeans(select(., -sample), na.rm = TRUE)) %>% 
  select(sample, row_mean) %>% 
  left_join(., sample_metadata[c(1,12)], by = join_by(sample == sample))%>%
  group_by(group1) %>% 
  summarise(mean=mean(row_mean, na.rm = TRUE))

bacteria_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column("sample") %>%
  mutate(
    row_mean = rowMeans(select(., -sample), na.rm = TRUE),
    row_median = apply(select(., -sample), 1, median, na.rm = TRUE)
  ) %>%
  select(sample, row_mean, row_median) %>%
  left_join(sample_metadata[c(1, 12)], by = join_by(sample == sample)) %>%
  group_by(group1) %>%
  summarise(
    group_mean_of_means = mean(row_mean, na.rm = TRUE),
    group_median_of_medians = median(row_median, na.rm = TRUE),
    group_mean_of_medians = mean(row_median, na.rm = TRUE)
  )

bacteria_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column("sample") %>%
  mutate(
    row_sum = rowSums(select(., -sample), na.rm = TRUE),
  ) %>%
  select(sample, row_sum) %>%
  left_join(sample_metadata[c(1, 12)], by = join_by(sample == sample)) %>%
  group_by(group1) %>%
  summarise(
    group_mean_of_means = mean(row_sum, na.rm = TRUE),
    group_median_of_medians = median(row_sum, na.rm = TRUE),
  )

```
```{r}
bacteria <- genome_metadata %>%
  filter(species=="Sarcina perfringens") %>%
  select(genome) %>%
  pull()

bacteria_counts <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% bacteria)

bacteria_means <- bacteria_counts %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)%>%
  rownames_to_column(., "sample") %>% 
  mutate(
    row_mean = rowMeans(select(., -sample), na.rm = TRUE),
    row_median = apply(select(., -sample), 1, median, na.rm = TRUE)
  ) %>% 
  select(sample, row_mean, row_median) %>%
  left_join(sample_metadata[c(1, 12)], by = join_by(sample == sample))

shapiro.test(bacteria_means$row_mean)
wilcox.test(row_mean ~ group1, data=bacteria_means)

shapiro.test(bacteria_means$row_median)
wilcox.test(row_median ~ group1, data=bacteria_means)
```

<!--chapter:end:Winter_summer.Rmd-->

# With wolves and dogs

```{r load_data_beta_wolves, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")

sample_metadata <- read_csv("data/sample_metadata.csv")
 
sample_metadata$individual <- sub("[0-9]{2}$", "", sample_metadata$animal)

sample_metadata_illu <- read_csv("data/metadata_illu.csv")
sample_metadata <- sample_metadata %>% 
  left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) 

load("data/beta.Rdata")
#samples to remove
samples_to_remove <- sample_metadata %>% 
  filter(Samples %in% c("Blank", "Saliva") | group2 %in% c("Unknown") | season=="winter" | !day %in% c("day_1", "Unknown") | individual == "GL_02_") %>% #
  select(sample) %>% 
  pull()

beta_q1n_mat <- as.matrix(beta_q1n$S)
beta_q1n_filtered <- beta_q1n_mat[!rownames(beta_q1n_mat) %in% samples_to_remove,
                        !colnames(beta_q1n_mat) %in% samples_to_remove]
beta_q1n <- as.dist(beta_q1n_filtered)

sample_metadata <- sample_metadata %>% 
  filter(!sample %in% samples_to_remove)
  #filter(individual != "GL_02_" | !Samples %in% c("Blank", "Saliva") | group2 != "Unknown" | season!="winter") 

genome_counts_filt <- genome_counts_filt %>% 
  select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome)
```

```{r gift_analysis, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
    select(genome, all_of(sample_metadata$sample)) %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

#Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

#Get community-weighed average GIFTs per sample
genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 
# genome_counts_row <- rownames_to_column(genome_counts_row, "genome")
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

**Neutral**
```{r}
neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")%>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  ggplot(aes(y = neutral, x = group2, group=group2, color=group2, fill=group2)) +
      geom_jitter(width = 0.3, show.legend = FALSE) +
      geom_boxplot(width = 0.7, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(y = "Alpha diversity (neutral)")

png(filename="alpha_all.png", width = 980, height = 580,)
plot(neutral)
dev.off()
```

```{r beta_div_neutral_wolves, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=20, fig.fullwidth=TRUE}
nmds_all1 <- beta_q1n %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(group1) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = group1)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +
  labs(color='Group')
# +
#   geom_text_repel(aes(label = individual), size=3)

png(filename="nmds1_all.png", width = 980, height = 780,)
plot(nmds_all1)
dev.off()
```



```{r permanova_q1_wolves, comment="", message=FALSE, warning=FALSE}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q1n), ]
sample_metadata <- sample_metadata_row %>% rownames_to_column("sample")

betadisper(beta_q1n, sample_metadata_row$group2) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n ~ group2, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_neutral_wolves, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(beta_q1n, sample_metadata$group2, perm = 999)
```

## Differential abundance
```{r chart1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
locationcolors=c('#c4d7d1','#408892','#c04062')

genome_counts_rel_fil<- genome_counts_filt %>% 
    select(one_of(c("genome",sample_metadata$sample))) %>% 
  column_to_rownames(., "genome")
  
genome_counts_rel_pa=1*(genome_counts_rel_fil>0)
#MAGrel_pa[1:6,1:6]
table_upset_analysis_cont=t(aggregate(t(genome_counts_rel_pa),by=list(sample_metadata$group2),FUN=sum)[,-1])
colnames(table_upset_analysis_cont)=levels(as.factor(sample_metadata$group2))
table_upset_analysis=(table_upset_analysis_cont>0)*1
table_upset_analysis=data.frame(table_upset_analysis)
table_upset_analysis=apply(table_upset_analysis,2,as.integer)
rownames(table_upset_analysis) <- rownames(genome_counts_rel_pa)

#pdf("figures/MAG_intersection.pdf",width=8,height=6, onefile=F)
upset(as.data.frame(table_upset_analysis),
  keep.order = T,
  sets = rev(c("Sled_dog","Dog","Wolf")),
  sets.bar.color= rev(locationcolors),
  mb.ratio = c(0.55, 0.45), order.by = "freq")
#dev.off()
```

```{r zero_phylo1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
phylo_samples <- sample_metadata %>%
  column_to_rownames("sample") %>%
  sample_data()
phylo_genome <- genome_counts_filt %>%
  select(one_of(c("genome", rownames(phylo_samples)))) %>%
  column_to_rownames("genome") %>%
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>%
  filter(genome %in% rownames(phylo_genome)) %>%
  mutate(genome2 = genome) %>% 
  column_to_rownames("genome2") %>%
  dplyr::select(domain, phylum, class, order, family, genus, species, genome) %>% 
  as.matrix() %>%
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r save_ancombc_summer, warning=FALSE, comments="", message=FALSE}
load("data/ancombc_results_groups.Rdata") #ancombc_results_wolves.Rdata
```

### MAG level
```{r ancom_rand_mag, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_mag = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = NULL,
  fix_formula = "group2",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "group2",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

### the global differential taxa
```{r}
res_global <- ancom_rand_output_mag$res_global
global_hits <- res_global %>%
  filter(diff_abn == TRUE) %>%
  select(taxon, W, p_val, q_val)
```

### Extract the results
#### Structural zeros

***Total number of structural zeros***
```{r}
# Extract structural zeros
structural_zeros <- ancom_rand_output_mag$zero_ind

# Convert to a data frame for better readability
structural_zeros_df <- as.data.frame(structural_zeros) %>% 
  column_to_rownames(., "taxon")

structural_zero_taxa <- rownames(structural_zeros_df)[apply(structural_zeros_df, 1, any)]

# Identify in which groups each taxon is a structural zero
structural_zero_groups <- structural_zeros_df[structural_zero_taxa, , drop = FALSE] %>% 
  rename("Dog"=1) %>% 
  rename("Sled_dog"=2) %>% 
  rename("Wolf"=3) 

# Total number of structural zeros
nrow(structural_zero_groups)

```

***Structural zeros in dogs***
```{r}
structural_zero_groups %>% 
  filter(Dog==TRUE) %>% 
  nrow()
```

***Structural zeros in sled dogs***
```{r}
structural_zero_groups %>% 
  filter(Sled_dog==TRUE) %>% 
  nrow()
```

***Structural zeros in wolves***
```{r}
structural_zero_groups %>% 
  filter(Wolf==TRUE) %>% 
  nrow()

structural_zero_groups %>% 
  filter(Wolf==TRUE) %>% 
  rownames_to_column(., "genome") %>% 
  left_join(., genome_metadata, by="genome") %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 
```


```{r labeling, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
 res_pair <- ancom_rand_output_mag$res_pair

# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_group2Sled_dog" ~ "Sled_dog vs Dog",
      contrast_key == "lfc_group2Wolf" ~ "Wolf vs Dog",
      contrast_key == "lfc_group2Wolf_group2Sled_dog" ~ "Wolf vs Sled_dog",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


```

```{r}
plot_data <- plot_data %>%
  left_join(genome_metadata %>% select(genome, phylum), by =join_by("taxon"=="genome"))

```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by = join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
  unique() %>%
  dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])#"Comparison:", 
  
  subset_data <- plot_data %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data$phylum)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data, aes(x = taxon_ordered, y = lfc, fill = phylum)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 6),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14, face = "bold"),
      legend.position = "right",
      legend.box = "vertical",
      plot.title = element_text(size = 18, face = "bold", hjust = 0.5)
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      #paste("Enriched Taxa:", c)
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Phylum"
    )
  
  ggsave(paste0("Fig_", gsub(" ", "_", c), ".png"), plot = p, width = 10, height = 25, dpi = 300)
}
```
### Phylum level
```{r ancom_rand_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_phylum = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = "phylum",
  fix_formula = "group2",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "group2",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

#### Structural zeros

***Total number of structural zeros***
```{r}
# Extract structural zeros
structural_zeros <- ancom_rand_output_phylum$zero_ind

# Convert to a data frame for better readability
structural_zeros_df <- as.data.frame(structural_zeros) %>% 
  column_to_rownames(., "taxon")

structural_zero_taxa <- rownames(structural_zeros_df)[apply(structural_zeros_df, 1, any)]

# Identify in which groups each taxon is a structural zero
structural_zero_groups <- structural_zeros_df[structural_zero_taxa, , drop = FALSE] %>% 
  rename("Dog"=1) %>% 
  rename("Sled_dog"=2) %>% 
  rename("Wolf"=3) 

# Total number of structural zeros
nrow(structural_zero_groups)

```

```{r  labeling_phylum, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
res_pair_phylum <- ancom_rand_output_phylum$res_pair
# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair_phylum), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data_phylum <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_phylum %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_group2Sled_dog" ~ "Sled_dog vs Dog",
      contrast_key == "lfc_group2Wolf" ~ "Wolf vs Dog",
      contrast_key == "lfc_group2Wolf_group2Sled_dog" ~ "Wolf vs Sled_dog",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )
plot_data_phylum
```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data_phylum$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)


for (c in unique_contrasts) {
  
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])# 

  
  subset_data_phylum <- plot_data_phylum %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data_phylum$taxon)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data_phylum, aes(x = taxon_ordered, y = lfc, fill = taxon)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(
        size = 0.5,
        linetype = "solid",
        colour = "black"
      ),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_blank(),
      legend.title = element_blank(),
      plot.title = element_text(size = 18, face = "bold", hjust = 0.5)
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      x = "Taxa",
      y = "Log-Fold Change",
    )
  
  ggsave(paste0("Fig_phylum_", gsub(" ", "_", c), ".png"), plot = p, width = 6, height = 8, dpi = 300)
}
```

### Genus level
```{r ancom_rand_genus, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_genus = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = "genus",
  fix_formula = "group2",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "group2",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r save_ancombc, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(ancom_rand_output_mag,
     ancom_rand_output_phylum,
     ancom_rand_output_genus,
     file = "data/ancombc_results_groups.Rdata")
```

#### Structural zeros

***Total number of structural zeros***
```{r}
# Extract structural zeros
structural_zeros <- ancom_rand_output_genus$zero_ind

# Convert to a data frame for better readability
structural_zeros_df <- as.data.frame(structural_zeros) %>% 
  column_to_rownames(., "taxon")

structural_zero_taxa <- rownames(structural_zeros_df)[apply(structural_zeros_df, 1, any)]

# Identify in which groups each taxon is a structural zero
structural_zero_groups <- structural_zeros_df[structural_zero_taxa, , drop = FALSE] %>% 
  rename("Dog"=1) %>% 
  rename("Sled_dog"=2) %>% 
  rename("Wolf"=3) 

# Total number of structural zeros
nrow(structural_zero_groups)

```

***Structural zeros in autumn***
```{r}
structural_zero_groups %>% 
  filter(Dog==TRUE) %>% 
  nrow()
```

***Structural zeros in wolves***
```{r}
structural_zero_groups %>% 
  filter(Sled_dog==TRUE) %>% 
  nrow()
```

***Structural zeros in wolves***
```{r}
structural_zero_groups %>% 
  filter(Wolf==TRUE) %>% 
  nrow()

structural_zero_groups %>% 
  filter(Wolf==TRUE) %>% 
  rownames_to_column(., "genome") %>%
  mutate(taxon_clean = sub(".*:", "", genome)) %>% 
  left_join(., genome_metadata %>% select(genus, family, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(phylum.x = coalesce(phylum.x, phylum.y)) %>% 
  select(-family,-phylum.y) %>% 
  rename(phylum=phylum.x) %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 
```
```{r labeling_genus, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
res_pair_genus <- ancom_rand_output_genus$res_pair
# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair_genus), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data_genus <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_genus %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_group2Sled_dog" ~ "Sled_dog vs Dog",
      contrast_key == "lfc_group2Wolf" ~ "Wolf vs Dog",
      contrast_key == "lfc_group2Wolf_group2Sled_dog" ~ "Wolf vs Sled_dog",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


```

```{r}
plot_data_genus <- plot_data_genus %>%
  mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
  left_join(., genome_metadata %>% select(genus, family, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(phylum.x = coalesce(phylum.x, phylum.y)) %>% 
  select(-family,-phylum.y) %>% 
  rename(phylum=phylum.x)

```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data_genus$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  titulu <- paste(locs[2], " - ", locs[1])
  
  subset_data_genus <- plot_data_genus %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) 
#  %>%  mutate(taxon_ordered = factor(taxon_clean, levels = unique(taxon_clean)))
  
  tax_color <- as.data.frame(unique(subset_data_genus$phylum)) %>% 
    rename(phylum=1) %>% 
    left_join(., colors_alphabetic, by="phylum")%>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()

  
  p <- ggplot(subset_data_genus, aes(x=lfc, y=forcats::fct_reorder(taxon_clean,lfc), fill=phylum))  +
    geom_bar(stat = "identity") +
 #   coord_flip() +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 6),
        axis.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical",
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))+   
      scale_fill_manual(values=tax_color) + 
    labs(
      title = titulu,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Direction"
    )
  
  ggsave(paste0("Fig_genus_", gsub(" ", "_", c), ".png"), plot = p, width = 6, height = 10, dpi = 300)
}
```

## Functional differences
## Groups MCI
```{r gitfs_functional_wild, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(group2) %>%
  summarise(MCI = mean(value), sd = sd(value)) %>%
  tt()
```
```{r gitfs_functional_wild, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(group1) %>%
  summarise(MCI = mean(value), sd = sd(value)) %>%
  tt()
```

```{r}
MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
kruskal.test(value ~ group2, data=MCI)
```

## Community elements differences:

```{r commun_wilcox_elem_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}

element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,7)], by="sample")

uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,locality), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ locality)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

selected_data <- element_gift %>%
  pivot_longer(-c(sample, locality), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_elements$trait)

# Now run Dunn's test per trait
library(rstatix)

pairwise_results <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ locality, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element)) %>%
  filter(!comparison %in% c("Mongolia vs South Africa", "Mongolia vs USA", "Laos vs Mongolia", "India vs Mongolia", "South Africa vs USA","Laos vs South Africa", "Laos vs India", "India vs South Africa", "India vs USA", "Laos vs USA"))

```

```{r function_plot_summer, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Locality Comparison",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )
```
```{r barplot_func_wolves, comment="", message=FALSE, warning=FALSE, fig.height=60, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,5,7)], by="sample")

plot_data <- element_gift %>%
  pivot_longer(-c(sample, locality, origin), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results$trait)%>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element)) 
# %>% 
#   filter(Function=="Xenobiotic degradation_Mercury")

# Plot boxplots per trait
ggplot(plot_data, aes(x = locality, y = value, fill = locality)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(aes(shape = origin), width = 0.2, size = 3) +
  #geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```

### by group

```{r commun_wilcox_elem_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,13)], by="sample")

uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,group2), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ group2)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

selected_data <- element_gift %>%
  pivot_longer(-c(sample, group2), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_elements$trait)

# Run Dunn's test per trait
pairwise_results <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ group2, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element))
```
```{r}
pairwise_results %>% group_by(comparison) %>%
  summarise(num_significant_functions = n()) %>% 
  arrange(num_significant_functions)
```
```{r function_plot_summer, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Groups",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )
```

```{r barplot_func_wolves, comment="", message=FALSE, warning=FALSE, fig.height=60, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,13)], by="sample")

plot_data <- element_gift %>%
  pivot_longer(-c(sample, group2), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results$trait)%>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element)) 
# %>% 
#   filter(Function=="Xenobiotic degradation_Mercury")

# Plot boxplots per trait
ggplot(plot_data, aes(x = group2, y = value, fill = group2)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "Trait Distributions Across Groups",
    x = "group",
    y = "Trait Value" )
```
```{r}
plot_data %>% group_by(group1, Function) %>% summarise(mean=mean(value)) %>% pivot_wider(names_from = group1, values_from = c(mean))
```


```{r}
pairwise_results %>% filter(comparison=="Dog vs Sled_dog_civilian_summer") %>% select(Function) %>% pull()
```

```{r barplot_func_wolves, comment="", message=FALSE, warning=FALSE, fig.height=60, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,5,12)], by="sample")%>% 
   filter(group1%in% c("Dog", "Sled_dog_civilian_summer"))

pairwise_dog_sled <- pairwise_results %>% filter(comparison=="Dog vs Sled_dog_civilian_summer")

plot_data <- element_gift %>%
  pivot_longer(-c(sample, group1, origin), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_dog_sled$trait)%>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element)) 
# %>% 
#   filter(Function=="Xenobiotic degradation_Mercury")

# Plot boxplots per trait
ggplot(plot_data, aes(x = group1, y = value, fill = group1)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(aes(shape = origin), width = 0.2, size = 3) +
  #geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "Trait Distributions Across Groups",
    x = "group",
    y = "Trait Value" )
```


## Community function differences:
```{r commun_wilcox_func_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,13)], by="sample")

uniqueFuncGIFT_db<- unique(GIFT_db[c(3,4,5)])

significant_functions <- function_gift %>%
    pivot_longer(-c(sample,group2), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ group2)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "functions")  %>% 
  select(-functions)

selected_data <- function_gift %>%
  pivot_longer(-c(sample, group2), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_functions$trait)

# Now run Dunn's test per trait
library(rstatix)

pairwise_results_func <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ group2, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function))

```

```{r pairwise_plot_func, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results_func, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Locality Comparison",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )

```

```{r barplot_func_wolves, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,13)], by="sample")

plot_data <- function_gift %>%
  pivot_longer(-c(sample, group2), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results_func$trait)%>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function)) 
# %>% 
#   filter(Function=="Xenobiotic degradation_Mercury")

# Plot boxplots per trait
ggplot(plot_data, aes(x = group2, y = value, fill = group2)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```


<!--chapter:end:Wolves_dogs.Rmd-->

