# Greenlandic sled dogs in summer

```{r knitr_options_beta, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(24022025)
```

## Filter data

```{r load_data_G_community, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

```{r load_data_host_filt_gr, comment="", message=FALSE, warning=FALSE}
sample_metadata <- read_csv("data/sample_metadata.csv")

 sample_metadata <- sample_metadata %>%
   filter(country== "Greenland")

# sample_metadata <- sample_metadata %>%
#   filter(country== "Greenland" | breed == "Wolf")

sample_metadata$individual <- sub("[0-9]{2}$", "", sample_metadata$animal)

sample_metadata_illu <- read_csv("data/metadata_illu.csv")
sample_metadata <- sample_metadata %>% 
  left_join(sample_metadata_illu[c(1,4,9)], by=join_by("sample"=="DogID")) %>% 
  filter(!Samples %in% c("Blank","Saliva")) %>% 
  filter(individual!="GL_02_") %>% 
  filter(season!="winter")

genome_counts_filt <- genome_counts_filt %>% 
    select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips
```


## Taxonomy overview 

**Number of bacteria phyla**
```{r phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**After merging all Bacillota**
```{r}
genome_metadata <- genome_metadata %>%
  mutate(phylum_clean = ifelse(grepl("^Bacillota_", phylum), "Bacillota", phylum))

genome_metadata %>% 
  filter(domain == "Bacteria")%>%
  dplyr::select(phylum_clean) %>%
  unique() %>%
  pull() %>%
  length()
```

### Stacked barplot

```{r taxonomy_G_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),  ~ . / sum(.)) %>%
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(., genome_metadata, by = join_by(genome == genome)) %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(count > 0) %>%
  ggplot(., aes(x = sample,
    y = count,
    fill = phylum,
    group = phylum
  )) +
  geom_bar(stat = "identity",
           colour = "white",
           linewidth = 0.1) +
  scale_fill_manual(values = phylum_colors) +
  facet_grid( ~ locality, scale = "free", space = "free") +
  guides(fill = guide_legend(ncol = 1)) +
  theme(
    axis.title.x = element_blank(),
    panel.background = element_blank(),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    strip.background = element_rect(fill = "white"),
    strip.text = element_text(size = 8, lineheight = 0.6),
    strip.placement = "outside",
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.line = element_line(
      linewidth = 0.5,
      linetype = "solid",
      colour = "black"
    )
  ) +
  labs(fill = "Phylum", y = "Relative abundance", x = "Samples")
```

### Phylum relative abundances

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum_clean,locality) %>%
  summarise(relabun=sum(count))
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum_clean) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    select(phylum_clean) %>%
    pull()
```
```{r taxonomy_phylum_summary_origin, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
  group_by(phylum_clean) %>%
  summarise(
    total_mean = mean(relabun * 100, na.rm = T),
    total_sd = sd(relabun * 100, na.rm = T),
    Daneborg_mean = mean(relabun[locality == "Daneborg"] * 100, na.rm =T),
    Daneborg_sd = sd(relabun[locality == "Daneborg"] * 100, na.rm =T),
    Disko_mean = mean(relabun[locality == "Disko"] * 100, na.rm = T),
    Disko_sd = sd(relabun[locality == "Disko"] * 100, na.rm =T),
    Illulissat_mean = mean(relabun[locality == "Illulissat"] * 100, na.rm = T),
    Illulissat_sd = sd(relabun[locality == "Illulissat"] * 100, na.rm = T),
    Ittoqqortoormiit_mean = mean(relabun[locality == "Ittoqqortoormiit"] * 100, na.rm = T),
    Ittoqqortoormiit_sd = sd(relabun[locality == "Ittoqqortoormiit"] * 100, na.rm = T),
    Wolf_mean = mean(relabun[locality == "Mongolia"] * 100, na.rm = T),
    Wolf_sd = sd(relabun[locality == "Mongolia"] * 100, na.rm = T)) %>%
  mutate(
    total = str_c(round(total_mean, 4), "±", round(total_sd, 3)),
    Daneborg = str_c(round(Daneborg_mean, 4), "±", round(Daneborg_sd, 3)),
    Disko = str_c(round(Disko_mean, 4), "±", round(Disko_sd, 3)),
    Illulissat = str_c(round(Illulissat_mean, 4), "±", round(Illulissat_sd, 3)),
    Ittoqqortoormiit = str_c(round(Ittoqqortoormiit_mean, 4), "±", round(Ittoqqortoormiit_sd, 3)),
    Wolf = str_c(round(Wolf_mean, 4), "±", round(Wolf_sd, 3))
  ) %>%
  arrange(-total_mean) %>%
  dplyr::select(phylum_clean, total, Daneborg, Disko,Illulissat,Ittoqqortoormiit, Wolf)
```



```{r alpha_G_div, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")


genome_counts_filt_beta <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt_beta <- genome_counts_filt_beta %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt_beta$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  filter(genome %in% rownames(dist)) %>% 
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))%>%
  full_join(functional, by = join_by(sample == sample))
```

### Plots

```{r alpha_div_G_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = locality, group=locality, color=locality, fill=locality)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10,angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Location", y = "Functional")
```

```{r div_plot_G, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3,plot4, ncol = 2))
```

### Values

```{r alpha_values_season, comment="", message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(alpha) %>%
  summarise(
    total_mean = mean(value, na.rm = T),
    total_sd = sd(value, na.rm = T),
    Daneborg_mean = mean(value[locality == "Daneborg"], na.rm =T),
    Daneborg_sd = sd(value[locality == "Daneborg"], na.rm =T),
    Disko_mean = mean(value[locality == "Disko"], na.rm = T),
    Disko_sd = sd(value[locality == "Disko"], na.rm =T),
    Illulissat_mean = mean(value[locality == "Illulissat"], na.rm = T),
    Illulissat_sd = sd(value[locality == "Illulissat"], na.rm = T),
    Ittoqqortoormiit_mean = mean(value[locality == "Ittoqqortoormiit"], na.rm = T),
    Ittoqqortoormiit_sd = sd(value[locality == "Ittoqqortoormiit"], na.rm = T),,
    Wolf_mean = mean(value[locality == "Mongolia"], na.rm = T),
    Wolf_sd = sd(value[locality == "Mongolia"], na.rm = T)) %>%
  mutate(
    total = str_c(round(total_mean, 3), "±", round(total_sd, 3)),
    Daneborg = str_c(round(Daneborg_mean, 3), "±", round(Daneborg_sd, 3)),
    Disko = str_c(round(Disko_mean, 3), "±", round(Disko_sd, 3)),
    Illulissat = str_c(round(Illulissat_mean, 3), "±", round(Illulissat_sd, 3)),
    Ittoqqortoormiit = str_c(round(Ittoqqortoormiit_mean, 3), "±", round(Ittoqqortoormiit_sd, 3)),
    Wolf = str_c(round(Wolf_mean, 3), "±", round(Wolf_sd, 3))
  ) %>%
  arrange(-total_mean) %>%
  dplyr::select(alpha,total, Daneborg, Disko,Illulissat,Ittoqqortoormiit, Wolf)

```

### Mixed models

```{r mixed_rich_prepost, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))

Model_richness <- MASS::glm.nb(richness ~ locality, data = alpha_div_meta,trace=TRUE)
anova(Model_richness)
summary(Model_richness)
emmeans(Model_richness, pairwise ~ locality)

Model_neutral<- lm(formula = neutral ~ locality, data = alpha_div_meta) 
anova(Model_neutral)
summary(Model_neutral)
emmeans(Model_neutral, pairwise ~ locality)

Model_phylo<- lm(formula = phylogenetic ~ locality, data = alpha_div_meta) 
anova(Model_phylo)
summary(Model_phylo)
emmeans(Model_phylo, pairwise ~ locality)

Model_functional<- lm(formula = functional ~ locality, data = alpha_div_meta) 
anova(Model_functional)
summary(Model_functional)
emmeans(Model_functional, pairwise ~ locality)
```

```{r beta_div_G, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

genome_counts <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts_filt$genome)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, tree = genome_tree)

genome_counts_filt_beta <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt_beta <- genome_counts_filt_beta %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt_beta$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

beta_q1f <- genome_counts_filt_beta %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, dist = dist)
```

```{r save_beta_G, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
save(beta_q0n, 
     beta_q1n, 
     beta_q1p,
     beta_q1f,
     file = "data/beta_greenland_wolf.Rdata")
```

### Beta diversity plots

```{r load_data_beta_summer, comment="", message=FALSE, warning=FALSE}
load("data/beta_greenland_summer.Rdata")
```

#### Richness

```{r beta_div_nmds_neutral_plot_G, comment="", message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +labs(color='Origin')
```

#### Neutral

```{r beta_div_neutral_G, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=20, fig.fullwidth=TRUE}
beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) +
  labs(color='Locality')
# +
#   geom_text_repel(aes(label = animal), size=3)

```


#### Phylogenetic

```{r beta_div_nmds_phylo1_plot_G, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
    geom_point(size = 4) +
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Locality')
```

#### Functional
```{r beta_div_nmds_func_plot_G, comment="", message=FALSE, warning=FALSE}
beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  mutate(x_cen = median(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = median(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = locality)) +
  geom_point(size = 4) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme_classic() +
  theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    ) +
    labs(color='Locality')
```

### Permanova analysis
```{r}
sample_metadata_row<- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q1n$S), ]
```

**Richness**
```{r permanova_locality_q0, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q0n$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$S ~ locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```

**Neutral**
```{r permanova_locality_q1, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$S ~ locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_neutral_locality, comment="", message=FALSE, warning=FALSE}
sample_metadata <- sample_metadata %>% arrange(match(sample,labels(beta_q1n$S)))
pairwise.adonis(beta_q1n$S, sample_metadata$locality, perm = 999)
```

**Phylogenetic**
```{r permanova_locality_qP, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1p$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1p$S ~ locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_phylo, comment="", message=FALSE, warning=FALSE}
sample_metadata <- sample_metadata %>% arrange(match(sample,labels(beta_q1p$S)))

pairwise.adonis(beta_q1p$S, sample_metadata$locality, perm = 999)
```

**Functional**
```{r permanova_locality_qF, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1f$S, sample_metadata_row$locality) %>% permutest(., pairwise = TRUE) 
adonis2(beta_q1f$S ~ place*locality, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999, by="terms") %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_functional, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(beta_q1f$S, sample_metadata$locality, perm = 999)
```



## Enrichment analysis: Ancombc2

```{r zero_phylo1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
phylo_samples <- sample_metadata %>%
  column_to_rownames("sample") %>%
  sample_data()
phylo_genome <- genome_counts_filt %>%
  select(one_of(c("genome", rownames(phylo_samples)))) %>%
  column_to_rownames("genome") %>%
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>%
  filter(genome %in% rownames(phylo_genome)) %>%
  mutate(genome2 = genome) %>% 
  column_to_rownames("genome2") %>%
  dplyr::select(domain, phylum, class, order, family, genus, species, genome) %>% 
  as.matrix() %>%
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r save_ancombc_summer, warning=FALSE, comments="", message=FALSE}
load("data/ancombc_results.Rdata") #ancombc_results_wolves.Rdata
```

### MAG level
```{r ancom_rand_mag, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_mag = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = NULL,
  fix_formula = "locality",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "locality",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

### the global differential taxa
```{r}
res_global <- ancom_rand_output_mag$res_global
global_hits <- res_global %>%
  filter(diff_abn == TRUE) %>%
  select(taxon, W, p_val, q_val)
```

### Extract the results
```{r}
 res_pair <- ancom_rand_output_mag$res_pair
```

```{r labeling, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}

# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


```

```{r}
plot_data <- plot_data %>%
  left_join(genome_metadata %>% select(genome, phylum), by =join_by("taxon"=="genome"))

```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by = join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
  unique() %>%
  dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])#"Comparison:", 
  
  subset_data <- plot_data %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data$phylum)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data, aes(x = taxon_ordered, y = lfc, fill = phylum)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 6),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14, face = "bold"),
      legend.position = "right",
      legend.box = "vertical",
      plot.title = element_text(size = 18, face = "bold", hjust = 0.5)
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      #paste("Enriched Taxa:", c)
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Phylum"
    )
  
  ggsave(paste0("Fig_", gsub(" ", "_", c), ".png"), plot = p, width = 10, height = 25, dpi = 300)
}
```

# top 10
```{r}
for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])

  subset_data <- plot_data %>%
    group_by(contrast) %>%
    filter(lfc > 0) %>%
    slice_max(order_by = lfc, n = 10) %>%
    bind_rows(
      plot_data %>%
        group_by(contrast) %>%
        filter(lfc < 0) %>%
        slice_min(order_by = lfc, n = 10)
    ) %>%
    ungroup() %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data$phylum)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data, aes(x = taxon_ordered, y = lfc, fill = phylum)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14, face = "bold"),
      legend.position = "right",
      legend.box = "vertical",
      plot.title = element_text(size = 18, face = "bold")
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Phylum"
    )
  
  ggsave(paste0("Fig_top10_", gsub(" ", "_", c), ".png"), plot = p, width = 8, height = 12, dpi = 300)
}
```

### Phylum level
```{r ancom_rand_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_phylum = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = "phylum",
  fix_formula = "locality",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "locality",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r}
res_pair_phylum <- ancom_rand_output_phylum$res_pair
```

```{r  labeling_phylum, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair_phylum), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data_phylum <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_phylum %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )
```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data_phylum$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)


for (c in unique_contrasts) {
  
  locs <- str_split(c, " vs ", simplify = TRUE)
  title <- paste(locs[2], " - ", locs[1])# 

  
  subset_data_phylum <- plot_data_phylum %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) %>%
    mutate(taxon_ordered = factor(taxon, levels = unique(taxon)))
  
  tax_color <- as.data.frame(unique(subset_data_phylum$taxon)) %>%
    rename(phylum = 1) %>%
    left_join(., colors_alphabetic, by = "phylum") %>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()
  
  p <- ggplot(subset_data_phylum, aes(x = taxon_ordered, y = lfc, fill = taxon)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    coord_flip() +
    theme(
      panel.background = element_blank(),
      axis.line = element_line(
        size = 0.5,
        linetype = "solid",
        colour = "black"
      ),
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 14, face = "bold"),
      legend.text = element_blank(),
      legend.title = element_blank(),
      plot.title = element_text(size = 18, face = "bold", hjust = 0.5)
    ) +
    scale_fill_manual(values = tax_color) +
    labs(
      title = title,
      x = "Taxa",
      y = "Log-Fold Change",
    )
  
  ggsave(paste0("Fig_phylum_", gsub(" ", "_", c), ".png"), plot = p, width = 6, height = 8, dpi = 300)
}
```

### Genus level
```{r ancom_rand_genus, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_genus = ancombc2(
  data = physeq_genome_filtered,
  assay_name = "counts",
  tax_level = "genus",
  fix_formula = "locality",
  p_adj_method = "holm",
  pseudo_sens = TRUE,
  prv_cut = 0,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "locality",
  struc_zero = TRUE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 2,
  verbose = TRUE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = FALSE,
  trend = FALSE,
  iter_control = list(
    tol = 1e-5,
    max_iter = 20,
    verbose = FALSE
  ),
  em_control = list(tol = 1e-5, max_iter = 100),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = NULL
)
```

```{r save_ancombc, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(ancom_rand_output_mag,
     ancom_rand_output_phylum,
     ancom_rand_output_genus,
     file = "data/ancombc_results.Rdata") #ancombc_results_wolves.Rdata
```


```{r}
res_pair_genus <- ancom_rand_output_genus$res_pair
```


```{r labeling_genus, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}

# Get log-fold change and q-value columns
lfc_cols <- grep("^lfc_", colnames(res_pair_genus), value = TRUE)
q_cols   <- gsub("^lfc_", "q_", lfc_cols)

# Build long table by pairing lfc and q for each contrast
plot_data_genus <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_genus %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


```

```{r}
plot_data_genus <- plot_data_genus %>%
  mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
  left_join(., genome_metadata %>% select(genus, family, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(phylum.x = coalesce(phylum.x, phylum.y)) %>% 
  select(-family,-phylum.y) %>% 
  rename(phylum=phylum.x)

```

```{r}
# Loop through each contrast
unique_contrasts <- unique(plot_data_genus$contrast)

taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  titulu <- paste(locs[2], " - ", locs[1])
  
  subset_data_genus <- plot_data_genus %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) 
#  %>%  mutate(taxon_ordered = factor(taxon_clean, levels = unique(taxon_clean)))
  
  tax_color <- as.data.frame(unique(subset_data_genus$phylum)) %>% 
    rename(phylum=1) %>% 
    left_join(., colors_alphabetic, by="phylum")%>%
    dplyr::arrange(phylum) %>%
    dplyr::select(colors) %>%
    pull()

  
  p <- ggplot(subset_data_genus, aes(x=lfc, y=forcats::fct_reorder(taxon_clean,lfc), fill=phylum))  +
    geom_bar(stat = "identity") +
 #   coord_flip() +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical",
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))+   
      scale_fill_manual(values=tax_color) + 
    labs(
      title = titulu,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Direction"
    )
  
  ggsave(paste0("Fig_genus_", gsub(" ", "_", c), ".png"), plot = p, width = 6, height = 10, dpi = 300)
}
```

```{r}
# Loop through each contrast
plot_data_genus <- lfc_cols %>%
  setNames(nm = .) %>%
  purrr::imap_dfr(function(lfc_col, name) {
    q_col <- gsub("^lfc_", "q_", lfc_col)
    
    res_pair_genus %>%
      select(taxon, !!lfc_col, !!q_col) %>%
      rename(
        lfc = !!lfc_col,
        q   = !!q_col
      ) %>%
      mutate(contrast_key = name)
  }) %>%
  filter(!is.na(lfc), !is.na(q), q < 0.05) %>%
  mutate(
    # Manually re-label contrasts for clarity
    contrast = case_when(
      contrast_key == "lfc_localityDisko" ~ "Disko vs Daneborg",
      contrast_key == "lfc_localityIllulissat" ~ "Illulissat vs Daneborg",
      contrast_key == "lfc_localityIttoqqortoormiit" ~ "Ittoqqortoormiit vs Daneborg",
      contrast_key == "lfc_localityMongolia" ~ "Mongolia vs Daneborg",
      contrast_key == "lfc_localityIllulissat_localityDisko" ~ "Illulissat vs Disko",
      contrast_key == "lfc_localityMongolia_localityDisko" ~ "Mongolia vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityDisko" ~ "Ittoqqortoormiit vs Disko",
      contrast_key == "lfc_localityIttoqqortoormiit_localityIllulissat" ~ "Ittoqqortoormiit vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIllulissat" ~ "Mongolia vs Illulissat",
      contrast_key == "lfc_localityMongolia_localityIttoqqortoormiit" ~ "Mongolia vs Ittoqqortoormiit",
      TRUE ~ contrast_key
    ),
    direction = ifelse(lfc > 0, paste("Enriched in", sub(" vs.*", "", contrast)),
                             paste("Enriched in", sub(".*vs ", "", contrast)))
  )


plot_data_genusF <- plot_data_genus %>%
  mutate(taxon_clean = sub(".*:", "", taxon)) %>% 
  left_join(., genome_metadata %>% select(genus, class, phylum) %>% distinct(), by =join_by("taxon_clean"=="genus"))%>% 
  left_join(., genome_metadata %>% select(family, class, phylum)%>% distinct(), by =join_by("taxon_clean"=="family")) %>%
  mutate(class.x = coalesce(class.x, class.y)) %>% 
  select(-class.y) %>% 
  rename(class=class.x)

unique_contrasts <- unique(plot_data_genus$contrast)

# taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
#   rownames_to_column(., "taxon")

# colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#   mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
#   right_join(taxonomy, by=join_by(phylum == phylum)) %>%
#   dplyr::select(phylum, colors) %>%
#   mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
# 	unique() %>%
# 	dplyr::arrange(phylum)

for (c in unique_contrasts) {
  locs <- str_split(c, " vs ", simplify = TRUE)
  titulu <- paste(locs[2], " - ", locs[1])
  
  subset_data_genusF <- plot_data_genusF %>%
    filter(contrast == c) %>%
    arrange(direction, lfc) 
#  %>%  mutate(taxon_ordered = factor(taxon_clean, levels = unique(taxon_clean)))
  
  # tax_color <- as.data.frame(unique(subset_data_genus$phylum)) %>% 
  #   rename(phylum=1) %>% 
  #   left_join(., colors_alphabetic, by="phylum")%>%
  #   dplyr::arrange(phylum) %>%
  #   dplyr::select(colors) %>%
  #   pull()

  
  p_f <- ggplot(subset_data_genusF, aes(x=lfc, y=forcats::fct_reorder(taxon_clean,lfc), fill=class))  +
    geom_bar(stat = "identity") +
 #   coord_flip() +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical",
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))+
    scale_fill_discrete()+
    labs(
      title = titulu,
      x = "Taxa",
      y = "Log-Fold Change",
      fill = "Faily"
    )
  
  ggsave(paste0("Fig_genus_color_family_", gsub(" ", "_", c), ".png"), plot = p_f, width = 10, height = 20, dpi = 300)
}
```

## Functional differences

```{r gift_analysis, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>% 
  select(genome, all_of(sample_metadata$sample)) %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

#Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

#Get community-weighed average GIFTs per sample
genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 
# genome_counts_row <- rownames_to_column(genome_counts_row, "genome")
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

## Groups MCI
```{r gitfs_functional_wild, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(locality) %>%
  summarise(MCI = mean(value), sd = sd(value)) %>%
  tt()
```

```{r}
MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
kruskal.test(value ~ locality, data=MCI)
```

## Community elements differences:
```{r comunity_elem_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,7)], by="sample")
```

```{r commun_wilcox_elem_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,locality), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ locality)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

selected_data <- element_gift %>%
  pivot_longer(-c(sample, locality), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_elements$trait)

# Now run Dunn's test per trait
library(rstatix)

pairwise_results <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ locality, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element))
```
```{r}
pairwise_results %>% group_by(comparison) %>%
  summarise(num_significant_functions = n()) %>% 
  arrange(num_significant_functions)
```
```{r function_plot_summer, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Locality Comparison",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )
```
```{r barplot_func, comment="", message=FALSE, warning=FALSE, fig.height=60, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,5,7)], by="sample")

plot_data <- element_gift %>%
  pivot_longer(-c(sample, locality, origin), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results$trait)%>% 
  left_join(., uniqueGIFT_db, by=join_by(trait==Code_element))

# Plot boxplots per trait
ggplot(plot_data, aes(x = locality, y = value, fill = locality)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(aes(shape = origin), width = 0.2, size = 3) +
  #geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```
```{r}
plot_data %>% group_by(locality, Function) %>% summarise(mean=mean(value)) %>% pivot_wider(names_from = locality, values_from = c(mean))
```
```{r}
plot_data %>% group_by(locality, Function) %>% summarise(mean=mean(value)) %>% pivot_wider(names_from = locality, values_from = c(mean)) %>% select(Function) %>% pull()
```

## Community function differences:
```{r comunity_func_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,7)], by="sample")
```

```{r commun_wilcox_func_summer, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueFuncGIFT_db<- unique(GIFT_db[c(3,4,5)])

significant_functions <- function_gift %>%
    pivot_longer(-c(sample,locality), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
  summarise(p_value = kruskal.test(value ~ locality)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "functions")  %>% 
  select(-functions)

selected_data <- function_gift %>%
  pivot_longer(-c(sample, locality), names_to = "trait", values_to = "value") %>%
  filter(trait %in% significant_functions$trait)

# Now run Dunn's test per trait
library(rstatix)

pairwise_results_func <- selected_data %>%
  group_by(trait) %>%
  dunn_test(value ~ locality, p.adjust.method = "BH")%>% 
  filter(p.adj<0.05) %>%
  mutate(comparison = paste(group1, "vs", group2)) %>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function))

```

```{r pairwise_plot_func, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
ggplot(pairwise_results_func, aes(x = comparison, y = fct_rev(Function))) +
  geom_tile(aes(fill = p.adj), color = "white") +
  geom_text(aes(label = p.adj.signif), size = 4, color = "black") +
  scale_fill_gradient(low = "#b30000", high = "#fef0d9" , name = "Adjusted p-value") +
  theme_minimal() +
  labs(
    title = "Significant Pairwise Comparisons by Trait",
    x = "Locality Comparison",
    y = "Trait"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )

```

```{r barplot_func_wolves, comment="", message=FALSE, warning=FALSE, fig.height=20, fig.width=30, fig.fullwidth=TRUE}
# Filter and reshape data for significant traits
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,5,7)], by="sample")

plot_data <- function_gift %>%
  pivot_longer(-c(sample, locality, origin), names_to = "trait", values_to = "value") %>%
  filter(trait %in% pairwise_results_func$trait)%>% 
  left_join(., uniqueFuncGIFT_db, by=join_by(trait==Code_function)) 
# %>% 
#   filter(Function=="Xenobiotic degradation_Mercury")

# Plot boxplots per trait
ggplot(plot_data, aes(x = locality, y = value, fill = locality)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(aes(shape = origin), width = 0.2, size = 3) +
  #geom_jitter(width = 0.2, size = 1, alpha = 0.4) +
  facet_wrap(~ Function, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(
    title = "Trait Distributions Across Localities",
    x = "Locality",
    y = "Trait Value" )
```



